<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:02 SGT 2011 -->
<TITLE>
TranscriptionEngine
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="TranscriptionEngine";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/TranscriptionEngine.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/transcription/TranscriptionEngine.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="TranscriptionEngine.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.biojava3.core.sequence.transcription</FONT>
<BR>
Class TranscriptionEngine</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../../resources/inherit.gif" ALT="extended by "><B>org.biojava3.core.sequence.transcription.TranscriptionEngine</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>TranscriptionEngine</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
Used as a way of encapsulating the data structures required to parse
 DNA to a Protein sequence.

 In order to build one look at  which provides
 intelligent defaults & allows you to build an engine which is nearly
 the same as the default one but with a few changes. All of the engine
 is customisable.

 By default the code will attempt to:

 <ul>
 <li>Trim Stops</li>
 <li>Convert initiating codons to M</li>
 <li>Allow for the fuzzy translation of Codons i.e. if it contains an
 N that produces a <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>&lt;&gt;
 with an X at that position
 </ul>
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>ayates</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This class is the way to create a <CODE>TranslationEngine</CODE>.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getAminoAcidCompounds()">getAminoAcidCompounds</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDefault()">getDefault</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default instance to use when Transcribing from DNA -&gt; RNA -&gt;
 Protein.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDnaCompounds()">getDnaCompounds</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDnaRnaTranslator()">getDnaRnaTranslator</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getProteinSequenceCreator()">getProteinSequenceCreator</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaAminoAcidTranslator()">getRnaAminoAcidTranslator</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaCompounds()">getRnaCompounds</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaSequenceCreator()">getRnaSequenceCreator</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getTable()">getTable</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>,<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.transcription.Frame...)">multipleFrameTranslation</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna,
                         <A HREF="../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>...&nbsp;frames)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A way of translating DNA in a number of frames</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#translate(org.biojava3.core.sequence.template.Sequence)">translate</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Quick method to let you go from a CDS to a Peptide quickly.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getDefault()"><!-- --></A><H3>
getDefault</H3>
<PRE>
public static <A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</A> <B>getDefault</B>()</PRE>
<DL>
<DD>Default instance to use when Transcribing from DNA -&gt; RNA -&gt;
 Protein. If you require anything that is not a default setting then look
 at  for customisation options.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="translate(org.biojava3.core.sequence.template.Sequence)"><!-- --></A><H3>
translate</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt; <B>translate</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna)</PRE>
<DL>
<DD>Quick method to let you go from a CDS to a Peptide quickly. It assumes
 you are translating only in the first frame
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>dna</CODE> - The CDS to translate
<DT><B>Returns:</B><DD>The Protein Sequence</DL>
</DD>
</DL>
<HR>

<A NAME="multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.transcription.Frame...)"><!-- --></A><H3>
multipleFrameTranslation</H3>
<PRE>
public java.util.Map&lt;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>,<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&gt; <B>multipleFrameTranslation</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna,
                                                                                 <A HREF="../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>...&nbsp;frames)</PRE>
<DL>
<DD>A way of translating DNA in a number of frames
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>dna</CODE> - The CDS to translate<DD><CODE>frames</CODE> - The Frames to translate in
<DT><B>Returns:</B><DD>All generated protein sequences in the given frames. Can have
 null entries</DL>
</DD>
</DL>
<HR>

<A NAME="getTable()"><!-- --></A><H3>
getTable</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A> <B>getTable</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getRnaAminoAcidTranslator()"><!-- --></A><H3>
getRnaAminoAcidTranslator</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</A> <B>getRnaAminoAcidTranslator</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getDnaRnaTranslator()"><!-- --></A><H3>
getDnaRnaTranslator</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</A> <B>getDnaRnaTranslator</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getProteinSequenceCreator()"><!-- --></A><H3>
getProteinSequenceCreator</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt; <B>getProteinSequenceCreator</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getRnaSequenceCreator()"><!-- --></A><H3>
getRnaSequenceCreator</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; <B>getRnaSequenceCreator</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getDnaCompounds()"><!-- --></A><H3>
getDnaCompounds</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; <B>getDnaCompounds</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getRnaCompounds()"><!-- --></A><H3>
getRnaCompounds</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; <B>getRnaCompounds</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAminoAcidCompounds()"><!-- --></A><H3>
getAminoAcidCompounds</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt; <B>getAminoAcidCompounds</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/TranscriptionEngine.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/transcription/TranscriptionEngine.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="TranscriptionEngine.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
