<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:01 SGT 2011 -->
<TITLE>
ChromosomeSequence
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ChromosomeSequence";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ChromosomeSequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/biojava3/core/sequence/ChromosomeSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ChromosomeSequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_org.biojava3.core.sequence.DNASequence">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.biojava3.core.sequence</FONT>
<BR>
Class ChromosomeSequence</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;
      <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">org.biojava3.core.sequence.DNASequence</A>
          <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>org.biojava3.core.sequence.ChromosomeSequence</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.lang.Iterable&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;, <A HREF="../../../../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</A>, <A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>ChromosomeSequence</B><DT>extends <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></DL>
</PRE>

<P>
A ChromosomeSequence is a DNASequence but keeps track of geneSequences
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Scooter Willis</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
</TABLE>
&nbsp;<A NAME="nested_classes_inherited_from_class_org.biojava3.core.sequence.DNASequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Nested classes/interfaces inherited from class org.biojava3.core.sequence.<A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</A></CODE></TD>
</TR>
</TABLE>
&nbsp;
<A NAME="nested_classes_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Nested classes/interfaces inherited from class org.biojava3.core.sequence.template.<A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A></CODE></TD>
</TR>
</TABLE>
&nbsp;

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#ChromosomeSequence()">ChromosomeSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Empty constructor used by tools that need a proper Bean that allows the actual
 sequence data to be set after construction.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#ChromosomeSequence(org.biojava3.core.sequence.template.SequenceReader)">ChromosomeSequence</A></B>(<A HREF="../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Fairly important constructor given the size of a ChromsomeSequence where the
 ProxySequenceReader could load from disk via RandomAccessFile so that the sequence
 doesn't need to be kept in memory.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#ChromosomeSequence(org.biojava3.core.sequence.template.SequenceReader, org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence</A></B>(<A HREF="../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader,
                   <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#ChromosomeSequence(java.lang.String)">ChromosomeSequence</A></B>(java.lang.String&nbsp;seqString)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;String is king and assume DNA</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#ChromosomeSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence</A></B>(java.lang.String&nbsp;seqString,
                   <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#addGene(org.biojava3.core.sequence.AccessionID, int, int, org.biojava3.core.sequence.Strand)">addGene</A></B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
        int&nbsp;bioBegin,
        int&nbsp;bioEnd,
        <A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#getChromosomeNumber()">getChromosomeNumber</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#getGene(java.lang.String)">getGene</A></B>(java.lang.String&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the gene based on accession.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.LinkedHashMap&lt;java.lang.String,<A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#getGeneSequences()">getGeneSequences</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the list of genes that have been added to the ChromosomeSequence where accession.toString is the key.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#removeGeneSequence(java.lang.String)">removeGeneSequence</A></B>(java.lang.String&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html#setChromosomeNumber(int)">setChromosomeNumber</A></B>(int&nbsp;chromosomeNumber)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.biojava3.core.sequence.DNASequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.biojava3.core.sequence.<A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getComplement()">getComplement</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getDNAType()">getDNAType</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getGCCount()">getGCCount</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getReverse()">getReverse</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getReverseComplement()">getReverseComplement</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence()">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.Frame)">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine, org.biojava3.core.sequence.transcription.Frame)">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#main(java.lang.String[])">main</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">setDNAType</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.biojava3.core.sequence.template.<A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(int, int, org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String, int)">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLength()">getLength</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer, java.lang.Integer, org.biojava3.core.sequence.Strand)">getSequenceAsString</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSource()">getSource</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer, java.lang.Integer)">getSubSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#iterator()">iterator</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#toString()">toString</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="ChromosomeSequence()"><!-- --></A><H3>
ChromosomeSequence</H3>
<PRE>
public <B>ChromosomeSequence</B>()</PRE>
<DL>
<DD>Empty constructor used by tools that need a proper Bean that allows the actual
 sequence data to be set after construction. Not recommended
<P>
</DL>
<HR>

<A NAME="ChromosomeSequence(java.lang.String)"><!-- --></A><H3>
ChromosomeSequence</H3>
<PRE>
public <B>ChromosomeSequence</B>(java.lang.String&nbsp;seqString)</PRE>
<DL>
<DD>String is king and assume DNA
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>seqString</CODE> - </DL>
</DL>
<HR>

<A NAME="ChromosomeSequence(org.biojava3.core.sequence.template.SequenceReader)"><!-- --></A><H3>
ChromosomeSequence</H3>
<PRE>
public <B>ChromosomeSequence</B>(<A HREF="../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader)</PRE>
<DL>
<DD>Fairly important constructor given the size of a ChromsomeSequence where the
 ProxySequenceReader could load from disk via RandomAccessFile so that the sequence
 doesn't need to be kept in memory. Could also be a NCBI proxy to load sequence
 data as needed from remote web server.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>proxyLoader</CODE> - </DL>
</DL>
<HR>

<A NAME="ChromosomeSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
ChromosomeSequence</H3>
<PRE>
public <B>ChromosomeSequence</B>(java.lang.String&nbsp;seqString,
                          <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</PRE>
<DL>
<DD>Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>seqString</CODE> - <DD><CODE>compoundSet</CODE> - </DL>
</DL>
<HR>

<A NAME="ChromosomeSequence(org.biojava3.core.sequence.template.SequenceReader, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
ChromosomeSequence</H3>
<PRE>
public <B>ChromosomeSequence</B>(<A HREF="../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader,
                          <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</PRE>
<DL>
<DD>Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>proxyLoader</CODE> - <DD><CODE>compoundSet</CODE> - </DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getChromosomeNumber()"><!-- --></A><H3>
getChromosomeNumber</H3>
<PRE>
public int <B>getChromosomeNumber</B>()</PRE>
<DL>
<DD><DL>

<DT><B>Returns:</B><DD>the chromosomeNumber</DL>
</DD>
</DL>
<HR>

<A NAME="setChromosomeNumber(int)"><!-- --></A><H3>
setChromosomeNumber</H3>
<PRE>
public void <B>setChromosomeNumber</B>(int&nbsp;chromosomeNumber)</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>chromosomeNumber</CODE> - the chromosomeNumber to set</DL>
</DD>
</DL>
<HR>

<A NAME="getGeneSequences()"><!-- --></A><H3>
getGeneSequences</H3>
<PRE>
public java.util.LinkedHashMap&lt;java.lang.String,<A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>&gt; <B>getGeneSequences</B>()</PRE>
<DL>
<DD>Get the list of genes that have been added to the ChromosomeSequence where accession.toString is the key.
 The list retains the order the genes are added
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="removeGeneSequence(java.lang.String)"><!-- --></A><H3>
removeGeneSequence</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A> <B>removeGeneSequence</B>(java.lang.String&nbsp;accession)</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="addGene(org.biojava3.core.sequence.AccessionID, int, int, org.biojava3.core.sequence.Strand)"><!-- --></A><H3>
addGene</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A> <B>addGene</B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
                            int&nbsp;bioBegin,
                            int&nbsp;bioEnd,
                            <A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</PRE>
<DL>
<DD>Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0. The
 GeneSequence that is returned will have a reference to parent chromosome sequence
 which actually contains the sequence data. Strand is important for positive and negative
 direction where negative strand means we need reverse complement. If negative strand then
 bioBegin will be greater than bioEnd
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - <DD><CODE>begin</CODE> - <DD><CODE>end</CODE> - <DD><CODE>strand</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getGene(java.lang.String)"><!-- --></A><H3>
getGene</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A> <B>getGene</B>(java.lang.String&nbsp;accession)</PRE>
<DL>
<DD>Get the gene based on accession. Will return null if not found
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ChromosomeSequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/biojava3/core/sequence/ChromosomeSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ChromosomeSequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_org.biojava3.core.sequence.DNASequence">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
