<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:01 SGT 2011 -->
<TITLE>
GeneSequence
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="GeneSequence";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GeneSequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/biojava3/core/sequence/GeneSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GeneSequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_org.biojava3.core.sequence.DNASequence">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.biojava3.core.sequence</FONT>
<BR>
Class GeneSequence</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;
      <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">org.biojava3.core.sequence.DNASequence</A>
          <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>org.biojava3.core.sequence.GeneSequence</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.lang.Iterable&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;, <A HREF="../../../../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</A>, <A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>GeneSequence</B><DT>extends <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></DL>
</PRE>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Scooter Willis</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
</TABLE>
&nbsp;<A NAME="nested_classes_inherited_from_class_org.biojava3.core.sequence.DNASequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Nested classes/interfaces inherited from class org.biojava3.core.sequence.<A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</A></CODE></TD>
</TR>
</TABLE>
&nbsp;
<A NAME="nested_classes_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Nested classes/interfaces inherited from class org.biojava3.core.sequence.template.<A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A></CODE></TD>
</TR>
</TABLE>
&nbsp;

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#GeneSequence(org.biojava3.core.sequence.ChromosomeSequence, int, int, org.biojava3.core.sequence.Strand)">GeneSequence</A></B>(<A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</A>&nbsp;parentSequence,
             int&nbsp;begin,
             int&nbsp;end,
             <A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A class that keeps track of the details of a GeneSequence which is difficult to properly model.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#addExon(org.biojava3.core.sequence.AccessionID, int, int)">addExon</A></B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
        int&nbsp;begin,
        int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add an ExonSequence mainly used to mark as a feature</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#addIntron(org.biojava3.core.sequence.AccessionID, int, int)">addIntron</A></B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
          int&nbsp;begin,
          int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add an Intron Currently used to mark an IntronSequence as a feature</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#addIntronsUsingExons()">addIntronsUsingExons</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Once everything has been added to the gene sequence where you might have added exon sequences only then you
 can infer the intron sequences and add them.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#addTranscript(org.biojava3.core.sequence.AccessionID, int, int)">addTranscript</A></B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
              int&nbsp;begin,
              int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add a transcription sequence to a gene which describes a ProteinSequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.ArrayList&lt;<A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getExonSequences()">getExonSequences</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the exons as an ArrayList</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.ArrayList&lt;<A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getIntronSequences()">getIntronSequences</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the introns as an ArrayList</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getLength()">getLength</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the length of the Sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getParentChromosomeSequence()">getParentChromosomeSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The parent ChromosomeSequence which contains the actual DNA sequence data</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getSequence5PrimeTo3Prime()">getSequence5PrimeTo3Prime</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction
 Returns the DNASequence representative of the 5' and 3' reading based on strand</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getStrand()">getStrand</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A gene should have Strand</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getTranscript(java.lang.String)">getTranscript</A></B>(java.lang.String&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the transcript sequence by accession</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.LinkedHashMap&lt;java.lang.String,<A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#getTranscripts()">getTranscripts</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the collection of transcription sequences assigned to this gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#removeExon(java.lang.String)">removeExon</A></B>(java.lang.String&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Remove the exon sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#removeIntron(java.lang.String)">removeIntron</A></B>(java.lang.String&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Remove the intron by accession</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#removeTranscript(java.lang.String)">removeTranscript</A></B>(java.lang.String&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Remove the transcript sequence from the gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/GeneSequence.html#setStrand(org.biojava3.core.sequence.Strand)">setStrand</A></B>(<A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.biojava3.core.sequence.DNASequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.biojava3.core.sequence.<A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getComplement()">getComplement</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getDNAType()">getDNAType</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getGCCount()">getGCCount</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getReverse()">getReverse</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getReverseComplement()">getReverseComplement</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence()">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.Frame)">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine, org.biojava3.core.sequence.transcription.Frame)">getRNASequence</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#main(java.lang.String[])">main</A>, <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">setDNAType</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.biojava3.core.sequence.template.<A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(int, int, org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String, int)">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer, java.lang.Integer, org.biojava3.core.sequence.Strand)">getSequenceAsString</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSource()">getSource</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer, java.lang.Integer)">getSubSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#iterator()">iterator</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#toString()">toString</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="GeneSequence(org.biojava3.core.sequence.ChromosomeSequence, int, int, org.biojava3.core.sequence.Strand)"><!-- --></A><H3>
GeneSequence</H3>
<PRE>
public <B>GeneSequence</B>(<A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</A>&nbsp;parentSequence,
                    int&nbsp;begin,
                    int&nbsp;end,
                    <A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</PRE>
<DL>
<DD>A class that keeps track of the details of a GeneSequence which is difficult to properly model. Two important concepts that is difficult
 to make everything flexible but still work. You can have GFF features that only describe Exons or Exons/Introns or CDS regions and one
 or more Transcriptions. You can have exon sequences but that does not imply transcription to the actual protein.

 The GeneSequence will keep track of Exons and Introns but to get a Protein sequence you need to start with a
 TranscriptSequence and then add CDS sequences.

 This is also a key class in the biojava-3-genome module for reading and writing GFF3 files
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>parentDNASequence</CODE> - <DD><CODE>begin</CODE> - <DD><CODE>end</CODE> - inclusive of end<DD><CODE>strand</CODE> - force a gene to have strand and transcription sequence will inherit</DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getParentChromosomeSequence()"><!-- --></A><H3>
getParentChromosomeSequence</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</A> <B>getParentChromosomeSequence</B>()</PRE>
<DL>
<DD>The parent ChromosomeSequence which contains the actual DNA sequence data
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getLength()"><!-- --></A><H3>
getLength</H3>
<PRE>
public int <B>getLength</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html#getLength()">Sequence</A></CODE></B></DD>
<DD>Returns the length of the Sequence
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html#getLength()">getLength</A></CODE> in interface <CODE><A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE><DT><B>Overrides:</B><DD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLength()">getLength</A></CODE> in class <CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="addIntronsUsingExons()"><!-- --></A><H3>
addIntronsUsingExons</H3>
<PRE>
public void <B>addIntronsUsingExons</B>()
                          throws java.lang.Exception</PRE>
<DL>
<DD>Once everything has been added to the gene sequence where you might have added exon sequences only then you
 can infer the intron sequences and add them. You may also have the case where you only added one or more
 TranscriptSequences and from that you can infer the exon sequences and intron sequences.
 Currently not implement
<P>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getStrand()"><!-- --></A><H3>
getStrand</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A> <B>getStrand</B>()</PRE>
<DL>
<DD>A gene should have Strand
<P>
<DD><DL>

<DT><B>Returns:</B><DD>the strand</DL>
</DD>
</DL>
<HR>

<A NAME="setStrand(org.biojava3.core.sequence.Strand)"><!-- --></A><H3>
setStrand</H3>
<PRE>
public void <B>setStrand</B>(<A HREF="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>strand</CODE> - the strand to set</DL>
</DD>
</DL>
<HR>

<A NAME="getTranscript(java.lang.String)"><!-- --></A><H3>
getTranscript</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A> <B>getTranscript</B>(java.lang.String&nbsp;accession)</PRE>
<DL>
<DD>Get the transcript sequence by accession
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getTranscripts()"><!-- --></A><H3>
getTranscripts</H3>
<PRE>
public java.util.LinkedHashMap&lt;java.lang.String,<A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A>&gt; <B>getTranscripts</B>()</PRE>
<DL>
<DD>Get the collection of transcription sequences assigned to this gene
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="removeTranscript(java.lang.String)"><!-- --></A><H3>
removeTranscript</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A> <B>removeTranscript</B>(java.lang.String&nbsp;accession)</PRE>
<DL>
<DD>Remove the transcript sequence from the gene
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="addTranscript(org.biojava3.core.sequence.AccessionID, int, int)"><!-- --></A><H3>
addTranscript</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A> <B>addTranscript</B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
                                        int&nbsp;begin,
                                        int&nbsp;end)
                                 throws java.lang.Exception</PRE>
<DL>
<DD>Add a transcription sequence to a gene which describes a ProteinSequence
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - <DD><CODE>begin</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD>
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE> - If the accession id is already used</DL>
</DD>
</DL>
<HR>

<A NAME="removeIntron(java.lang.String)"><!-- --></A><H3>
removeIntron</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A> <B>removeIntron</B>(java.lang.String&nbsp;accession)</PRE>
<DL>
<DD>Remove the intron by accession
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="addIntron(org.biojava3.core.sequence.AccessionID, int, int)"><!-- --></A><H3>
addIntron</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A> <B>addIntron</B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
                                int&nbsp;begin,
                                int&nbsp;end)
                         throws java.lang.Exception</PRE>
<DL>
<DD>Add an Intron Currently used to mark an IntronSequence as a feature
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - <DD><CODE>begin</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD>
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="removeExon(java.lang.String)"><!-- --></A><H3>
removeExon</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</A> <B>removeExon</B>(java.lang.String&nbsp;accession)</PRE>
<DL>
<DD>Remove the exon sequence
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="addExon(org.biojava3.core.sequence.AccessionID, int, int)"><!-- --></A><H3>
addExon</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</A> <B>addExon</B>(<A HREF="../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession,
                            int&nbsp;begin,
                            int&nbsp;end)
                     throws java.lang.Exception</PRE>
<DL>
<DD>Add an ExonSequence mainly used to mark as a feature
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - <DD><CODE>begin</CODE> - <DD><CODE>end</CODE> - 
<DT><B>Returns:</B><DD>
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getExonSequences()"><!-- --></A><H3>
getExonSequences</H3>
<PRE>
public java.util.ArrayList&lt;<A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</A>&gt; <B>getExonSequences</B>()</PRE>
<DL>
<DD>Get the exons as an ArrayList
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getIntronSequences()"><!-- --></A><H3>
getIntronSequences</H3>
<PRE>
public java.util.ArrayList&lt;<A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A>&gt; <B>getIntronSequences</B>()</PRE>
<DL>
<DD>Get the introns as an ArrayList
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getSequence5PrimeTo3Prime()"><!-- --></A><H3>
getSequence5PrimeTo3Prime</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A> <B>getSequence5PrimeTo3Prime</B>()</PRE>
<DL>
<DD>Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction
 Returns the DNASequence representative of the 5' and 3' reading based on strand
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GeneSequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/biojava3/core/sequence/GeneSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GeneSequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_org.biojava3.core.sequence.DNASequence">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
