<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:01 SGT 2011 -->
<TITLE>
RNASequence
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="RNASequence";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/RNASequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/biojava3/core/sequence/RNASequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="RNASequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.biojava3.core.sequence</FONT>
<BR>
Class RNASequence</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;
      <IMG SRC="../../../../resources/inherit.gif" ALT="extended by "><B>org.biojava3.core.sequence.RNASequence</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.lang.Iterable&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;, <A HREF="../../../../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</A>, <A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>RNASequence</B><DT>extends <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</DL>
</PRE>

<P>
RNASequence where RNACompoundSet are the allowed values
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Scooter Willis <willishf at gmail dot com></DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
</TABLE>
&nbsp;<A NAME="nested_classes_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Nested classes/interfaces inherited from class org.biojava3.core.sequence.template.<A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A></CODE></TD>
</TR>
</TABLE>
&nbsp;

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#RNASequence(org.biojava3.core.sequence.template.ProxySequenceReader)">RNASequence</A></B>(<A HREF="../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a RNA aequence from a proxy reader</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#RNASequence(org.biojava3.core.sequence.template.ProxySequenceReader, org.biojava3.core.sequence.template.CompoundSet)">RNASequence</A></B>(<A HREF="../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader,
            <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a RNA sequence from a proxy reader and user defined RNA compound set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#RNASequence(java.lang.String)">RNASequence</A></B>(java.lang.String&nbsp;seqString)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a RNA sequence from a String</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#RNASequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">RNASequence</A></B>(java.lang.String&nbsp;seqString,
            <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a RNA sequence from a string with a user defined RNA compound set</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#getComplement()">getComplement</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the complement view of the RNA sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#getGC()">getGC</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#getInverse()">getInverse</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the inverse view of the sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#getProteinSequence()">getProteinSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the ProteinSequence from the RNA sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getProteinSequence</A></B>(<A HREF="../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</A>&nbsp;engine)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the ProteinSequene from the RNA sequence with user defined
 transcription engine</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../org/biojava3/core/sequence/RNASequence.html#getReverseComplement()">getReverseComplement</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get reverse complement view of the sequence</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.biojava3.core.sequence.template.<A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(int, int, org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String, int)">getFeatures</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLength()">getLength</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer, java.lang.Integer, org.biojava3.core.sequence.Strand)">getSequenceAsString</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSource()">getSource</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer, java.lang.Integer)">getSubSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#iterator()">iterator</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</A>, <A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#toString()">toString</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="RNASequence(java.lang.String)"><!-- --></A><H3>
RNASequence</H3>
<PRE>
public <B>RNASequence</B>(java.lang.String&nbsp;seqString)</PRE>
<DL>
<DD>Create a RNA sequence from a String
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>seqString</CODE> - </DL>
</DL>
<HR>

<A NAME="RNASequence(org.biojava3.core.sequence.template.ProxySequenceReader)"><!-- --></A><H3>
RNASequence</H3>
<PRE>
public <B>RNASequence</B>(<A HREF="../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader)</PRE>
<DL>
<DD>Create a RNA aequence from a proxy reader
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>proxyLoader</CODE> - </DL>
</DL>
<HR>

<A NAME="RNASequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
RNASequence</H3>
<PRE>
public <B>RNASequence</B>(java.lang.String&nbsp;seqString,
                   <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</PRE>
<DL>
<DD>Create a RNA sequence from a string with a user defined RNA compound set
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>seqString</CODE> - <DD><CODE>compoundSet</CODE> - </DL>
</DL>
<HR>

<A NAME="RNASequence(org.biojava3.core.sequence.template.ProxySequenceReader, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
RNASequence</H3>
<PRE>
public <B>RNASequence</B>(<A HREF="../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader,
                   <A HREF="../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</PRE>
<DL>
<DD>Create a RNA sequence from a proxy reader and user defined RNA compound set
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>proxyLoader</CODE> - <DD><CODE>compoundSet</CODE> - </DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getReverseComplement()"><!-- --></A><H3>
getReverseComplement</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; <B>getReverseComplement</B>()</PRE>
<DL>
<DD>Get reverse complement view of the sequence
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getInverse()"><!-- --></A><H3>
getInverse</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; <B>getInverse</B>()</PRE>
<DL>
<DD>Get the inverse view of the sequence. It is the reverse sequence from
 end to begin where use reverse could imply complement. Called getInverse()
 in the hopes of making less confusing.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html#getInverse()">getInverse</A></CODE> in interface <CODE><A HREF="../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE><DT><B>Overrides:</B><DD><CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</A></CODE> in class <CODE><A HREF="../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getComplement()"><!-- --></A><H3>
getComplement</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; <B>getComplement</B>()</PRE>
<DL>
<DD>Get the complement view of the RNA sequence
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getProteinSequence()"><!-- --></A><H3>
getProteinSequence</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A> <B>getProteinSequence</B>()</PRE>
<DL>
<DD>Get the ProteinSequence from the RNA sequence
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)"><!-- --></A><H3>
getProteinSequence</H3>
<PRE>
public <A HREF="../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A> <B>getProteinSequence</B>(<A HREF="../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</A>&nbsp;engine)</PRE>
<DL>
<DD>Get the ProteinSequene from the RNA sequence with user defined
 transcription engine
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>engine</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getGC()"><!-- --></A><H3>
getGC</H3>
<PRE>
public double <B>getGC</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/RNASequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/biojava3/core/sequence/RNASequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="RNASequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_classes_inherited_from_class_org.biojava3.core.sequence.template.AbstractSequence">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
