<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:03 SGT 2011 -->
<TITLE>
T-Index
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="T-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-19.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-21.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-20.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-20.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>
<A NAME="_T_"><!-- --></A><H2>
<B>T</B></H2>
<DL>
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription"><B>Table</B></A> - Interface in <A HREF="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A><DD>Provides a way of separating us from the specific <CODE>IUPACTable</CODE> even
 though this is the only implementing class for the interface.<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#table(java.lang.Integer)"><B>table(Integer)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>Uses the static instance of <A HREF="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io"><CODE>IUPACParser</CODE></A> to find instances of
 <A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><CODE>IUPACParser.IUPACTable</CODE></A>s by ID.
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#table(java.lang.String)"><B>table(String)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>Uses the static instance of <A HREF="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io"><CODE>IUPACParser</CODE></A> to find instances of
 <A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><CODE>IUPACParser.IUPACTable</CODE></A>s by its String name
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#table(org.biojava3.core.sequence.transcription.Table)"><B>table(Table)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription"><B>Table.CaseInsensitiveTriplet</B></A> - Class in <A HREF="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A><DD>Class used to hold three nucleotides together and allow for equality
 to be assessed in a case insensitive manner.<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#Table.CaseInsensitiveTriplet(org.biojava3.core.sequence.compound.NucleotideCompound, org.biojava3.core.sequence.compound.NucleotideCompound, org.biojava3.core.sequence.compound.NucleotideCompound)"><B>Table.CaseInsensitiveTriplet(NucleotideCompound, NucleotideCompound, NucleotideCompound)</B></A> - 
Constructor for class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription"><B>Table.Codon</B></A> - Class in <A HREF="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A><DD>Instance of a Codon which is 3 <A HREF="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><CODE>NucleotideCompound</CODE></A>s, its
 corresponding <A HREF="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><CODE>AminoAcidCompound</CODE></A> and if it is a start or stop codon.<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#Table.Codon(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet, org.biojava3.core.sequence.compound.AminoAcidCompound, boolean, boolean)"><B>Table.Codon(Table.CaseInsensitiveTriplet, AminoAcidCompound, boolean, boolean)</B></A> - 
Constructor for class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#Table.Codon(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet)"><B>Table.Codon(Table.CaseInsensitiveTriplet)</B></A> - 
Constructor for class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence"><B>TaxonomyID</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>A sequence can be associated with a species or Taxonomy ID<DT><A HREF="../org/biojava3/core/sequence/TaxonomyID.html#TaxonomyID(java.lang.String, org.biojava3.core.sequence.DataSource)"><B>TaxonomyID(String, DataSource)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html#tearDown()"><B>tearDown()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html" title="class in org.biojava3.core.sequence.io">FastaReaderTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html#tearDown()"><B>tearDown()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParserTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html#tearDownClass()"><B>tearDownClass()</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html" title="class in org.biojava3.core.sequence.io">FastaReaderTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html#tearDownClass()"><B>tearDownClass()</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParserTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundTest.html#testAmbiguity()"><B>testAmbiguity()</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundTest.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundTest.html#testBasicAmbiguity()"><B>testBasicAmbiguity()</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundTest.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/LocationTest.html#testBasicCircularLocation()"><B>testBasicCircularLocation()</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/LocationTest.html" title="class in org.biojava3.core.sequence.location">LocationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceViewTest.html#testGetCompoundAt()"><B>testGetCompoundAt()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceViewTest.html" title="class in org.biojava3.core.sequence">SequenceViewTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignmentTest.html#testGetCompoundsAt()"><B>testGetCompoundsAt()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignmentTest.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignmentTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceViewTest.html#testInverse()"><B>testInverse()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceViewTest.html" title="class in org.biojava3.core.sequence">SequenceViewTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceViewTest.html#testLastIndexOf()"><B>testLastIndexOf()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceViewTest.html" title="class in org.biojava3.core.sequence">SequenceViewTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html#testParseHeader()"><B>testParseHeader()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParserTest</A>
<DD>GenBank                           gi|gi-number|gb|accession|locus
 ENA Data Library                 gi|gi-number|emb|accession|locus
 DDBJ, DNA Database of Japan       gi|gi-number|dbj|accession|locus
 NBRF PIR                          pir||entry
 Protein Research Foundation       prf||name
 SWISS-PROT                        sp|accession|name
 Brookhaven Protein Data Bank (1)  pdb|entry|chain
 Brookhaven Protein Data Bank (2)  entry:chain|PDBID|CHAIN|SEQUENCE
 PDB EBI                           PDB:1ECY_A mol:protein length:142  ECOTIN
 Patents                           pat|country|number
 GenInfo Backbone Id               bbs|number
 General database identifier       gnl|database|identifier
 NCBI Reference Sequence           ref|accession|locus
 Local Sequence identifier         lcl|identifier
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html#testProcess()"><B>testProcess()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html" title="class in org.biojava3.core.sequence.io">FastaReaderTest</A>
<DD>Test of process method, of class FastaReader.
<DT><A HREF="../org/biojava3/core/sequence/location/LocationTest.html#testStrandFlip()"><B>testStrandFlip()</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/LocationTest.html" title="class in org.biojava3.core.sequence.location">LocationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/LocationTest.html#testSubLocations()"><B>testSubLocations()</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/LocationTest.html" title="class in org.biojava3.core.sequence.location">LocationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/LocationTest.html#testWithStrandSwitch()"><B>testWithStrandSwitch()</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/LocationTest.html" title="class in org.biojava3.core.sequence.location">LocationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/TextFeature.html" title="class in org.biojava3.core.sequence.features"><B>TextFeature</B></A>&lt;<A HREF="../org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">S</A> extends <A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</A>&gt;,<A HREF="../org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</A><DD>A implmentation of AbstractFeature<DT><A HREF="../org/biojava3/core/sequence/features/TextFeature.html#TextFeature(java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><B>TextFeature(String, String, String, String)</B></A> - 
Constructor for class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/TextFeature.html" title="class in org.biojava3.core.sequence.features">TextFeature</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)"><B>toList(Sequence&lt;C&gt;)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>For the given <A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> this will return a <CODE>List</CODE> filled with
 the Compounds of that <A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>.
<DT><A HREF="../org/biojava3/core/sequence/AccessionID.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#toString(int)"><B>toString(int)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</A>
<DD>Get a string representation of the MSA with a fixed width
<DT><A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#toString(org.biojava3.core.sequence.template.LightweightProfile.StringFormat)"><B>toString(LightweightProfile.StringFormat)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</A>
<DD>Support for different MSA formats
<DT><A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</A>
<DD>String representation of the MSA
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/LightweightProfile.html#toString()"><B>toString()</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</A>
<DD>Returns a simple view of the alignment profile.
<DT><A HREF="../org/biojava3/core/sequence/template/LightweightProfile.html#toString(int)"><B>toString(int)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</A>
<DD>Returns a formatted view of the alignment profile.
<DT><A HREF="../org/biojava3/core/sequence/template/LightweightProfile.html#toString(org.biojava3.core.sequence.template.LightweightProfile.StringFormat)"><B>toString(LightweightProfile.StringFormat)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</A>
<DD>Returns a formatted view of the alignment profile.
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#toString(org.biojava3.core.sequence.template.Sequence)"><B>toString(Sequence&lt;C&gt;)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>Shortcut to <A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)"><CODE>SequenceMixin.toStringBuilder(org.biojava3.core.sequence.template.Sequence)</CODE></A>
 which calls toString() on the resulting object.
<DT><A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/CRC64Checksum.html#toString()"><B>toString()</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</A>
<DD>Returns a zero-padded 16 character wide string containing the current
 value of this checksum in uppercase hexadecimal format.
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)"><B>toStringBuilder(Sequence&lt;C&gt;)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>For the given Sequence this will return a <CODE>StringBuilder</CODE> object
 filled with the results of <CODE>Compound#toString()</CODE>.
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription"><B>TranscriptionEngine</B></A> - Class in <A HREF="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A><DD>Used as a way of encapsulating the data structures required to parse
 DNA to a Protein sequence.<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription"><B>TranscriptionEngine.Builder</B></A> - Class in <A HREF="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A><DD>This class is the way to create a <CODE>TranslationEngine</CODE>.<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#TranscriptionEngine.Builder()"><B>TranscriptionEngine.Builder()</B></A> - 
Constructor for class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence"><B>TranscriptSequence</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>This is the sequence if you want to go from a gene sequence to a protein sequence.<DT><A HREF="../org/biojava3/core/sequence/TranscriptSequence.html#TranscriptSequence(org.biojava3.core.sequence.GeneSequence, int, int)"><B>TranscriptSequence(GeneSequence, int, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#translate(F)"><B>translate(F)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/CompoundTranslator.html#translate(F)"><B>translate(F)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#translate(org.biojava3.core.sequence.template.Sequence)"><B>translate(Sequence&lt;NucleotideCompound&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</A>
<DD>Quick method to let you go from a CDS to a Peptide quickly.
<DT><A HREF="../org/biojava3/core/sequence/TranslationTest.html#translateBrca2()"><B>translateBrca2()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranslationTest.html" title="class in org.biojava3.core.sequence">TranslationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/TranslationTest.html#translateBrca2ExonOne()"><B>translateBrca2ExonOne()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranslationTest.html" title="class in org.biojava3.core.sequence">TranslationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/TranslationTest.html#translateInitMet()"><B>translateInitMet()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranslationTest.html" title="class in org.biojava3.core.sequence">TranslationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/TranslationTest.html#translateInternalStops()"><B>translateInternalStops()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranslationTest.html" title="class in org.biojava3.core.sequence">TranslationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#translateMany(F)"><B>translateMany(F)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/CompoundTranslator.html#translateMany(F)"><B>translateMany(F)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/TranslationTest.html#translateN()"><B>translateN()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranslationTest.html" title="class in org.biojava3.core.sequence">TranslationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#translateNCodons()"><B>translateNCodons()</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</A>
<DD>Indicates if we want to force exact translation of compounds or not i.e.
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#translateNCodons(boolean)"><B>translateNCodons(boolean)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#translateToRna()"><B>translateToRna()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions"><B>TranslationException</B></A> - Exception in <A HREF="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</A><DD>Thrown from AbstractCompundTranslator<DT><A HREF="../org/biojava3/core/exceptions/TranslationException.html#TranslationException(java.lang.String)"><B>TranslationException(String)</B></A> - 
Constructor for exception org.biojava3.core.exceptions.<A HREF="../org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/exceptions/TranslationException.html#TranslationException(java.lang.Exception)"><B>TranslationException(Exception)</B></A> - 
Constructor for exception org.biojava3.core.exceptions.<A HREF="../org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/exceptions/TranslationException.html#TranslationException(java.lang.String, java.lang.Exception)"><B>TranslationException(String, Exception)</B></A> - 
Constructor for exception org.biojava3.core.exceptions.<A HREF="../org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/TranslationTest.html" title="class in org.biojava3.core.sequence"><B>TranslationTest</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/TranslationTest.html#TranslationTest()"><B>TranslationTest()</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranslationTest.html" title="class in org.biojava3.core.sequence">TranslationTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#trimStop(boolean)"><B>trimStop(boolean)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#twoBit()"><B>twoBit()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage"><B>TwoBitSequenceReader</B></A>&lt;<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</A> extends <A HREF="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A><DD>Implementation of the 2bit encoding.<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(org.biojava3.core.sequence.template.Sequence)"><B>TwoBitSequenceReader(Sequence&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)"><B>TwoBitSequenceReader(String, CompoundSet&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.AccessionID)"><B>TwoBitSequenceReader(String, CompoundSet&lt;C&gt;, AccessionID)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(org.biojava3.core.sequence.storage.TwoBitSequenceReader.TwoBitArrayWorker)"><B>TwoBitSequenceReader(TwoBitSequenceReader.TwoBitArrayWorker&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(org.biojava3.core.sequence.storage.TwoBitSequenceReader.TwoBitArrayWorker, org.biojava3.core.sequence.AccessionID)"><B>TwoBitSequenceReader(TwoBitSequenceReader.TwoBitArrayWorker&lt;C&gt;, AccessionID)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage"><B>TwoBitSequenceReader.TwoBitArrayWorker</B></A>&lt;<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</A> extends <A HREF="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A><DD>Extension of the BitArrayWorker which provides the 2bit implementation
 code.<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int)"><B>TwoBitSequenceReader.TwoBitArrayWorker(CompoundSet&lt;C&gt;, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int[])"><B>TwoBitSequenceReader.TwoBitArrayWorker(CompoundSet&lt;C&gt;, int[])</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(org.biojava3.core.sequence.template.Sequence)"><B>TwoBitSequenceReader.TwoBitArrayWorker(Sequence&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)"><B>TwoBitSequenceReader.TwoBitArrayWorker(String, CompoundSet&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-19.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-21.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-20.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-20.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>

</BODY>
</HTML>
