<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:03 SGT 2011 -->
<TITLE>
S-Index
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="S-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-18.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-20.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-19.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-19.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>
<A NAME="_S_"><!-- --></A><H2>
<B>S</B></H2>
<DL>
<DT><A HREF="../org/biojava3/core/util/Hashcoder.html#SEED"><B>SEED</B></A> - 
Static variable in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</A>
<DD>An initial value for a <code>hashCode</code>, to which we add
 contributions from fields.
<DT><A HREF="../org/biojava3/core/util/XMLHelper.html#selectElements(org.w3c.dom.Element, java.lang.String)"><B>selectElements(Element, String)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/XMLHelper.html#selectParentElement(org.w3c.dom.Element, java.lang.String)"><B>selectParentElement(Element, String)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/XMLHelper.html#selectSingleElement(org.w3c.dom.Element, java.lang.String)"><B>selectSingleElement(Element, String)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#seqArraySize(int)"><B>seqArraySize(int)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><B>Sequence</B></A>&lt;<A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Interface in <A HREF="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A><DD>Main interface for defining a collection of Compounds and accessing them
 using biological indexes<DT><A HREF="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage"><B>SequenceAsStringHelper</B></A>&lt;<A HREF="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A><DD>This is a common method that can be used across multiple storage/proxy implementations to
 handle Negative strand and other interesting elements of sequence data.<DT><A HREF="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html#SequenceAsStringHelper()"><B>SequenceAsStringHelper()</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence"><B>SequenceComparator</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>Used to sort sequences<DT><A HREF="../org/biojava3/core/sequence/SequenceComparator.html#SequenceComparator()"><B>SequenceComparator()</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence">SequenceComparator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template"><B>SequenceCreatorInterface</B></A>&lt;<A HREF="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Interface in <A HREF="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#sequenceEquality()"><B>sequenceEquality()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)"><B>sequenceEquality(Sequence&lt;C&gt;, Sequence&lt;C&gt;)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>A case-sensitive manner of comparing two sequence objects together.
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)"><B>sequenceEqualityIgnoreCase(Sequence&lt;C&gt;, Sequence&lt;C&gt;)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>A case-insensitive manner of comparing two sequence objects together.
<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader"><B>SequenceFileProxyLoader</B></A>&lt;<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A><DD>This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#SequenceFileProxyLoader(java.io.File, org.biojava3.core.sequence.io.template.SequenceParserInterface, long, int, org.biojava3.core.sequence.template.CompoundSet)"><B>SequenceFileProxyLoader(File, SequenceParserInterface, long, int, CompoundSet&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions"><B>SequenceLengthError</B></A> - Error in <A HREF="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</A><DD>Currently not used<DT><A HREF="../org/biojava3/core/exceptions/SequenceLengthError.html#SequenceLengthError(java.lang.String)"><B>SequenceLengthError(String)</B></A> - 
Constructor for error org.biojava3.core.exceptions.<A HREF="../org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions">SequenceLengthError</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location"><B>SequenceLocation</B></A>&lt;<A HREF="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">S</A> extends <A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</A>&gt;,<A HREF="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>A location in a sequence that keeps a reference to its parent sequence<DT><A HREF="../org/biojava3/core/sequence/location/SequenceLocation.html#SequenceLocation(int, int, S)"><B>SequenceLocation(int, int, S)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template"><B>SequenceMixin</B></A> - Class in <A HREF="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A><DD>Provides a set of static methods to be used as static imports when needed
 across multiple Sequence implementations but inheritance gets in the way.<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#SequenceMixin()"><B>SequenceMixin()</B></A> - 
Constructor for class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template"><B>SequenceMixin.SequenceIterator</B></A>&lt;<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A><DD>A basic sequence iterator which iterates over the given Sequence by
 biological index.<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#SequenceMixin.SequenceIterator(org.biojava3.core.sequence.template.Sequence)"><B>SequenceMixin.SequenceIterator(Sequence&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence"><B>SequenceOptimizationHints</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>A static class that provides optimization hints for memory or performance handling of sequence data.<DT><A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.html#SequenceOptimizationHints()"><B>SequenceOptimizationHints()</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava3.core.sequence"><B>SequenceOptimizationHints.SequenceCollection</B></A> - Enum in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence"><B>SequenceOptimizationHints.SequenceUsage</B></A> - Enum in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template"><B>SequenceParserInterface</B></A> - Interface in <A HREF="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template"><B>SequenceProxyView</B></A>&lt;<A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html#SequenceProxyView()"><B>SequenceProxyView()</B></A> - 
Constructor for class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html#SequenceProxyView(org.biojava3.core.sequence.template.Sequence)"><B>SequenceProxyView(Sequence&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html#SequenceProxyView(org.biojava3.core.sequence.template.Sequence, java.lang.Integer, java.lang.Integer)"><B>SequenceProxyView(Sequence&lt;C&gt;, Integer, Integer)</B></A> - 
Constructor for class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</A>
<DD>Main constructor for working with SequenceProxyViews
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template"><B>SequenceReader</B></A>&lt;<A HREF="../org/biojava3/core/sequence/template/SequenceReader.html" title="type parameter in SequenceReader">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Interface in <A HREF="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util"><B>SequenceTools</B></A> - Class in <A HREF="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/util/SequenceTools.html#SequenceTools()"><B>SequenceTools()</B></A> - 
Constructor for class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template"><B>SequenceView</B></A>&lt;<A HREF="../org/biojava3/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Interface in <A HREF="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/SequenceViewTest.html" title="class in org.biojava3.core.sequence"><B>SequenceViewTest</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/SequenceViewTest.html#SequenceViewTest()"><B>SequenceViewTest()</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceViewTest.html" title="class in org.biojava3.core.sequence">SequenceViewTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)"><B>setAccession(AccessionID)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)"><B>setAnnotationType(AbstractSequence.AnnotationType)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)"><B>setBioBegin(Integer)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)"><B>setBioEnd(Integer)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html#setBioEnd(java.lang.Integer)"><B>setBioEnd(Integer)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html#setBioStart(java.lang.Integer)"><B>setBioStart(Integer)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setChildrenFeatures(java.util.List)"><B>setChildrenFeatures(List&lt;FeatureInterface&lt;S, C&gt;&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>Set the children features
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setChildrenFeatures(java.util.List)"><B>setChildrenFeatures(List&lt;FeatureInterface&lt;S, C&gt;&gt;)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>Set the children features
<DT><A HREF="../org/biojava3/core/sequence/ChromosomeSequence.html#setChromosomeNumber(int)"><B>setChromosomeNumber(int)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(char, int)"><B>setCompoundAt(char, int)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</A>
<DD>Converts from char to Compound and sets it at the given biological index
<DT><A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C, int)"><B>setCompoundAt(C, int)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</A>
<DD>Sets the compound at the specified biological index
<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</A>
<DD>Class is immutable & so this is unsupported
<DT><A HREF="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</A>
<DD>Unsupported
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;C&gt;)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><B>setCompoundSet(CompoundSet&lt;NucleotideCompound&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.views.<A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</A>
<DD>Once the sequence is retrieved set the contents and make sure everything this is valid
<DT><A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setContents(java.util.List)"><B>setContents(List&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</A>
<DD>Class is immutable & so this is unsupported
<DT><A HREF="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</A>
<DD>Unsupported
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceReader.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html#setContents(java.lang.String)"><B>setContents(String)</B></A> - 
Method in class org.biojava3.core.sequence.views.<A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setDatabase(java.lang.String)"><B>setDatabase(String)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)"><B>setDatabaseReferences(DatabaseReferenceInterface)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>Set the description that can be used to describe the feature
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/Compound.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#setDescription(java.lang.String)"><B>setDescription(String)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)"><B>setDNAType(DNASequence.DNAType)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)"><B>setFeaturesKeyWord(FeaturesKeyWordInterface)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setId(java.lang.String)"><B>setId(String)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setInitialized(boolean)"><B>setInitialized(boolean)</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaGeneWriter.html#setLineLength(int)"><B>setLineLength(int)</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaWriter.html#setLineLength(int)"><B>setLineLength(int)</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaWriter.html#setLineSeparator(java.lang.String)"><B>setLineSeparator(String)</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</A>
<DD>Allow an override of operating system line separator for programs that needs a specific CRLF or CR or LF option
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setLocation(org.biojava3.core.sequence.location.SequenceLocation)"><B>setLocation(SequenceLocation&lt;S, C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>A feature could be a single sequence position like a mutation or a post translational modification of an amino acid.
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setLocation(org.biojava3.core.sequence.location.SequenceLocation)"><B>setLocation(SequenceLocation&lt;S, C&gt;)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>The new location for this feature.
<DT><A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html#setLongName(java.lang.String)"><B>setLongName(String)</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html#setLongName(java.lang.String)"><B>setLongName(String)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/Compound.html#setLongName(java.lang.String)"><B>setLongName(String)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setLongName(java.lang.String)"><B>setLongName(String)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#setLongName(java.lang.String)"><B>setLongName(String)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html#setMolecularWeight(java.lang.Float)"><B>setMolecularWeight(Float)</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html#setMolecularWeight(java.lang.Float)"><B>setMolecularWeight(Float)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/Compound.html#setMolecularWeight(java.lang.Float)"><B>setMolecularWeight(Float)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setMolecularWeight(java.lang.Float)"><B>setMolecularWeight(Float)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#setMolecularWeight(java.lang.Float)"><B>setMolecularWeight(Float)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)"><B>setNotesList(ArrayList&lt;String&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)"><B>setOriginalHeader(String)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/ProteinSequence.html#setParentDNASequence(org.biojava3.core.sequence.template.AbstractSequence, java.lang.Integer, java.lang.Integer)"><B>setParentDNASequence(AbstractSequence, Integer, Integer)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A>
<DD>A Protein sequence can be stand alone or loaded from a transcript sequence.
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setParentFeature(org.biojava3.core.sequence.features.FeatureInterface)"><B>setParentFeature(FeatureInterface&lt;S, C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>A feature can be the child or contained by a parent feature.
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setParentFeature(org.biojava3.core.sequence.features.FeatureInterface)"><B>setParentFeature(FeatureInterface&lt;S, C&gt;)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>Set the parent feature
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)"><B>setParentSequence(AbstractSequence&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setProperties(java.util.LinkedHashMap)"><B>setProperties(LinkedHashMap&lt;String, String&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)"><B>setProxySequenceReader(SequenceReader&lt;C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>Very important method that allows external mappings of sequence data and features.
<DT><A HREF="../org/biojava3/core/sequence/features/QuantityFeature.html#setQuantities(java.util.List)"><B>setQuantities(List&lt;Number&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.html#setSequenceCollection(org.biojava3.core.sequence.SequenceOptimizationHints.SequenceCollection)"><B>setSequenceCollection(SequenceOptimizationHints.SequenceCollection)</B></A> - 
Static method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)"><B>setSequenceScore(Double)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.html#setSequenceUsage(org.biojava3.core.sequence.SequenceOptimizationHints.SequenceUsage)"><B>setSequenceUsage(SequenceOptimizationHints.SequenceUsage)</B></A> - 
Static method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setShortDescription(java.lang.String)"><B>setShortDescription(String)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setShortDescription(java.lang.String)"><B>setShortDescription(String)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>Set the short description that can be used to describe the feature
<DT><A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html#setShortName(java.lang.String)"><B>setShortName(String)</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html#setShortName(java.lang.String)"><B>setShortName(String)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/Compound.html#setShortName(java.lang.String)"><B>setShortName(String)</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setShortName(java.lang.String)"><B>setShortName(String)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#setShortName(java.lang.String)"><B>setShortName(String)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setSource(java.lang.String)"><B>setSource(String)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>Set the feature source
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setSource(java.lang.String)"><B>setSource(String)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>Change the source of the FeatureInterface.
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)"><B>setSource(String)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>Added support for the source of this sequence for GFF3 export
<DT><A HREF="../org/biojava3/core/sequence/GeneSequence.html#setStrand(org.biojava3.core.sequence.Strand)"><B>setStrand(Strand)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)"><B>setTaxonomy(TaxonomyID)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPool(java.util.concurrent.ThreadPoolExecutor)"><B>setThreadPool(ThreadPoolExecutor)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Sets thread pool to any given <CODE>ThreadPoolExecutor</CODE> to allow use of an alternative execution style.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolCPUsAvailable(int)"><B>setThreadPoolCPUsAvailable(int)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Sets thread pool to reserve a given number of processor cores for foreground or other use.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolCPUsFraction(float)"><B>setThreadPoolCPUsFraction(float)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Sets thread pool to a given fraction of the available processors.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolDefault()"><B>setThreadPoolDefault()</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Sets thread pool to default of 1 background thread for each processor core.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolSingle()"><B>setThreadPoolSingle()</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Sets thread pool to a single background thread.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolSize(int)"><B>setThreadPoolSize(int)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Sets thread pool to given size.
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setType(java.lang.String)"><B>setType(String)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>Set the feature type
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setType(java.lang.String)"><B>setType(String)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>Change the type of this feature.
<DT><A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setUniprotbaseURL(java.lang.String)"><B>setUniprotbaseURL(String)</B></A> - 
Static method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setUniprotDirectoryCache(java.lang.String)"><B>setUniprotDirectoryCache(String)</B></A> - 
Static method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html#setUp()"><B>setUp()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html" title="class in org.biojava3.core.sequence.io">FastaReaderTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html#setUp()"><B>setUp()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParserTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignmentTest.html#setup()"><B>setup()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignmentTest.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignmentTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html#setUpClass()"><B>setUpClass()</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderTest.html" title="class in org.biojava3.core.sequence.io">FastaReaderTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html#setUpClass()"><B>setUpClass()</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/GenericFastaHeaderParserTest.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParserTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)"><B>setUserCollection(Collection&lt;Object&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html#setUserObject(java.lang.Object)"><B>setUserObject(Object)</B></A> - 
Method in class org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</A>
<DD>Allow the user to associate an object with the feature.
<DT><A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html#setUserObject(java.lang.Object)"><B>setUserObject(Object)</B></A> - 
Method in interface org.biojava3.core.sequence.features.<A HREF="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#shuffle(org.biojava3.core.sequence.template.Sequence)"><B>shuffle(Sequence&lt;C&gt;)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>Implements sequence shuffling by first materializing the given
 <A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> into a <CODE>List</CODE>, applying
 <CODE>Collections.shuffle(List)</CODE> and then returning the shuffled
 elements in a new instance of <CODE>SequenceBackingStore</CODE> which behaves
 as a <A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#shutdown()"><B>shutdown()</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Disables new tasks from being submitted and closes the thread pool cleanly.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#shutdownAndAwaitTermination()"><B>shutdownAndAwaitTermination()</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Closes the thread pool.
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location"><B>SimpleLocation</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>Very basic implementation of the Location interface which defines a series
 of simple constructors.<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(int, int)"><B>SimpleLocation(int, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point)"><B>SimpleLocation(Point, Point)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(int, int, org.biojava3.core.sequence.Strand)"><B>SimpleLocation(int, int, Strand)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand)"><B>SimpleLocation(Point, Point, Strand)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, org.biojava3.core.sequence.AccessionID)"><B>SimpleLocation(Point, Point, Strand, AccessionID)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, org.biojava3.core.sequence.AccessionID)"><B>SimpleLocation(Point, Point, Strand, boolean, AccessionID)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, boolean)"><B>SimpleLocation(Point, Point, Strand, boolean, boolean)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(int, int, org.biojava3.core.sequence.Strand, org.biojava3.core.sequence.location.template.Location...)"><B>SimpleLocation(int, int, Strand, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, org.biojava3.core.sequence.location.template.Location...)"><B>SimpleLocation(Point, Point, Strand, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, org.biojava3.core.sequence.location.template.Location...)"><B>SimpleLocation(Point, Point, Strand, boolean, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, java.util.List)"><B>SimpleLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html#SimpleLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, boolean, java.util.List)"><B>SimpleLocation(Point, Point, Strand, boolean, boolean, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location"><B>SimplePoint</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>Basic implementation of the Point interface.<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#SimplePoint(int)"><B>SimplePoint(int)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#SimplePoint(int, boolean, boolean)"><B>SimplePoint(int, boolean, boolean)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#singleCompoundSequence()"><B>singleCompoundSequence()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage"><B>SingleCompoundSequenceReader</B></A>&lt;<A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A><DD>An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).<DT><A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#SingleCompoundSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet, int)"><B>SingleCompoundSequenceReader(String, CompoundSet&lt;C&gt;, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</A>
<DD>Public constructor to be used with String based constructor
<DT><A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#SingleCompoundSequenceReader(C, org.biojava3.core.sequence.template.CompoundSet, int)"><B>SingleCompoundSequenceReader(C, CompoundSet&lt;C&gt;, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</A>
<DD>Build the object with a compound rather than a String
<DT><A HREF="../org/biojava3/core/util/FlatFileCache.html#size()"><B>size()</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/SoftHashMap.html#size()"><B>size()</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#skip(long)"><B>skip(long)</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</A>
<DD>Skips characters.
<DT><A HREF="../org/biojava3/core/util/UncompressInputStream.html#skip(long)"><B>skip(long)</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util"><B>SoftHashMap</B></A>&lt;<A HREF="../org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">K</A>,<A HREF="../org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">V</A>&gt; - Class in <A HREF="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</A><DD>A in memory cache using soft references.<DT><A HREF="../org/biojava3/core/util/SoftHashMap.html#SoftHashMap()"><B>SoftHashMap()</B></A> - 
Constructor for class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/SoftHashMap.html#SoftHashMap(int)"><B>SoftHashMap(int)</B></A> - 
Constructor for class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence"><B>StartCodonSequence</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>Used to map the start codon feature on a gene<DT><A HREF="../org/biojava3/core/sequence/StartCodonSequence.html#StartCodonSequence(org.biojava3.core.sequence.TranscriptSequence, int, int)"><B>StartCodonSequence(TranscriptSequence, int, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence"><B>StopCodonSequence</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>Used to map the stop codon sequence on a gene<DT><A HREF="../org/biojava3/core/sequence/StopCodonSequence.html#StopCodonSequence(org.biojava3.core.sequence.TranscriptSequence, int, int)"><B>StopCodonSequence(TranscriptSequence, int, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence"><B>Strand</B></A> - Enum in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>Provides a way of representing the strand of a sequence, location
 hit or feature.<DT><A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader"><B>StringProxySequenceReader</B></A>&lt;<A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A><DD>An example of a ProxySequenceReader that is created from a String.<DT><A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#StringProxySequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)"><B>StringProxySequenceReader(String, CompoundSet&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#submit(java.util.concurrent.Callable, java.lang.String)"><B>submit(Callable&lt;T&gt;, String)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Queues up a task and adds a log entry.
<DT><A HREF="../org/biojava3/core/util/ConcurrencyTools.html#submit(java.util.concurrent.Callable)"><B>submit(Callable&lt;T&gt;)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</A>
<DD>Queues up a task and adds a default log entry.
<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#subSequence()"><B>subSequence()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/EditSequenceTest.html#substitute()"><B>substitute()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/EditSequenceTest.html" title="class in org.biojava3.core.sequence">EditSequenceTest</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-18.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-20.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-19.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-19.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>

</BODY>
</HTML>
