<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:03 SGT 2011 -->
<TITLE>
I-Index
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="I-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-8.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-10.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-9.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-9.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>
<A NAME="_I_"><!-- --></A><H2>
<B>I</B></H2>
<DL>
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence, C)"><B>indexOf(Sequence&lt;C&gt;, C)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>Performs a linear search of the given Sequence for the given compound.
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#initMet(boolean)"><B>initMet(boolean)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util"><B>InputStreamProvider</B></A> - Class in <A HREF="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</A><DD>A class that provides an InputStream from a File.<DT><A HREF="../org/biojava3/core/util/InputStreamProvider.html#InputStreamProvider()"><B>InputStreamProvider()</B></A> - 
Constructor for class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/XMLHelper.html#inputStreamToDocument(java.io.InputStream)"><B>inputStreamToDocument(InputStream)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.html" title="class in org.biojava3.core.sequence.location"><B>InsdcLocations</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>A collection of locations which are used whenever we work with INSDC; some
 of which could be deprecated (from INSDC's point of view) yet appear
 in records.<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.html#InsdcLocations()"><B>InsdcLocations()</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.html" title="class in org.biojava3.core.sequence.location">InsdcLocations</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html" title="class in org.biojava3.core.sequence.location"><B>InsdcLocations.BondLocation</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>Used to represent bond locations equivalent to bond(7,8) or bond(7).<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html#InsdcLocations.BondLocation(org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.BondLocation(Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.BondLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html#InsdcLocations.BondLocation(java.util.List)"><B>InsdcLocations.BondLocation(List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.BondLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location"><B>InsdcLocations.GroupLocation</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>Deprecated in INSDC yet still appears; equivalent to the order()
 directive except no 5' to 3' ordering is defined.<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#InsdcLocations.GroupLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.GroupLocation(Point, Point, Strand, boolean, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#InsdcLocations.GroupLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.GroupLocation(Point, Point, Strand, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#InsdcLocations.GroupLocation(int, int, org.biojava3.core.sequence.Strand, org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.GroupLocation(int, int, Strand, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#InsdcLocations.GroupLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, java.util.List)"><B>InsdcLocations.GroupLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#InsdcLocations.GroupLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, java.util.List)"><B>InsdcLocations.GroupLocation(Point, Point, Strand, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#InsdcLocations.GroupLocation(int, int, org.biojava3.core.sequence.Strand, java.util.List)"><B>InsdcLocations.GroupLocation(int, int, Strand, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava3.core.sequence.location"><B>InsdcLocations.OneOfLocation</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>Deprecated in INSDC; refers to a set of locations of which one
 location could be valid e.g.<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html#InsdcLocations.OneOfLocation(org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.OneOfLocation(Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OneOfLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html#InsdcLocations.OneOfLocation(java.util.List)"><B>InsdcLocations.OneOfLocation(List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OneOfLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location"><B>InsdcLocations.OrderLocation</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>Used to describe a 5' to 3' ordering but no firm assurance it is correct<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#InsdcLocations.OrderLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.OrderLocation(Point, Point, Strand, boolean, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#InsdcLocations.OrderLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.OrderLocation(Point, Point, Strand, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#InsdcLocations.OrderLocation(int, int, org.biojava3.core.sequence.Strand, org.biojava3.core.sequence.location.template.Location...)"><B>InsdcLocations.OrderLocation(int, int, Strand, Location...)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#InsdcLocations.OrderLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, boolean, java.util.List)"><B>InsdcLocations.OrderLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#InsdcLocations.OrderLocation(org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.location.template.Point, org.biojava3.core.sequence.Strand, java.util.List)"><B>InsdcLocations.OrderLocation(Point, Point, Strand, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#InsdcLocations.OrderLocation(int, int, org.biojava3.core.sequence.Strand, java.util.List)"><B>InsdcLocations.OrderLocation(int, int, Strand, List&lt;Location&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location"><B>InsdcParser</B></A> - Class in <A HREF="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A><DD>Parser for working with INSDC style locations.<DT><A HREF="../org/biojava3/core/sequence/location/InsdcParser.html#InsdcParser()"><B>InsdcParser()</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/InsdcParser.html#InsdcParser(org.biojava3.core.sequence.DataSource)"><B>InsdcParser(DataSource)</B></A> - 
Constructor for class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/EditSequenceTest.html#insert()"><B>insert()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/EditSequenceTest.html" title="class in org.biojava3.core.sequence">EditSequenceTest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence"><B>IntronSequence</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/IntronSequence.html#IntronSequence(org.biojava3.core.sequence.GeneSequence, int, int)"><B>IntronSequence(GeneSequence, int, int)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A>
<DD>Place holder for Intron sequence features
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#intValue()"><B>intValue()</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</A>
<DD>Attempts to provide an int version of this codon which multiplies
 each position by
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html#inverse(org.biojava3.core.sequence.template.Sequence)"><B>inverse(Sequence&lt;C&gt;)</B></A> - 
Static method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</A>
<DD>A method which attempts to do the right thing when is comes to a
 reverse/reverse complement
<DT><A HREF="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util"><B>IOUtils</B></A> - Class in <A HREF="../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/io/util/IOUtils.html#IOUtils()"><B>IOUtils()</B></A> - 
Constructor for class org.biojava3.core.sequence.io.util.<A HREF="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html" title="interface in org.biojava3.core.sequence.io.util"><B>IOUtils.ReaderProcessor</B></A> - Interface in <A HREF="../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A><DD>Closure interface used when working with
 <CODE>IOUtils#processReader(String)</CODE>.<DT><A HREF="../org/biojava3/aaproperties/IPeptideProperties.html" title="interface in org.biojava3.aaproperties"><B>IPeptideProperties</B></A> - Interface in <A HREF="../org/biojava3/aaproperties/package-summary.html">org.biojava3.aaproperties</A><DD>TODO
 Create adaptor methods for all methodsChange to Char instead of AminoAcidCompound or AminoAcid
 Implement all the methods
 
 An interface to generate some basic physico-chemical properties of protein sequences.<br/>
 The following properties could be generated:
 <p/>
 Molecular weight<br/>
 Extinction coefficient<br/>
 Instability index<br/>
 Apliphatic index<br/>
 Average hydropathy value<br/>
 Isoelectric point<br/>
 Net charge at pH 7<br/>
 Composition of specified amino acid<br/>
 Composition of the 20 standard amino acid<br/><DT><A HREF="../org/biojava3/core/sequence/compound/NucleotideCompound.html#isAmbiguous()"><B>isAmbiguous()</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html#isBetweenCompounds()"><B>isBetweenCompounds()</B></A> - 
Method in class org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Location.html#isBetweenCompounds()"><B>isBetweenCompounds()</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</A>
<DD>Returns true if the position is meant to represent a point between
 two points such as 78^79.
<DT><A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html#isCircular()"><B>isCircular()</B></A> - 
Method in class org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Location.html#isCircular()"><B>isCircular()</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</A>
<DD>Indicates if this location is circular.
<DT><A HREF="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#isComplementable()"><B>isComplementable()</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#isComplementable()"><B>isComplementable()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html#isComplementable()"><B>isComplementable()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</A>
<DD>NucleotideCompounds can always complement
<DT><A HREF="../org/biojava3/core/sequence/template/CompoundSet.html#isComplementable()"><B>isComplementable()</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html#isComplex()"><B>isComplex()</B></A> - 
Method in class org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Location.html#isComplex()"><B>isComplex()</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</A>
<DD>Returns true if the location is considered to be complex meaning
 the location is actually composed of sub-locations.
<DT><A HREF="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#isCompoundStringLengthEqual()"><B>isCompoundStringLengthEqual()</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#isCompoundStringLengthEqual()"><B>isCompoundStringLengthEqual()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/CompoundSet.html#isCompoundStringLengthEqual()"><B>isCompoundStringLengthEqual()</B></A> - 
Method in interface org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>
<DD>Returns true if all String representations of Compounds are of the
 same length.
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#isHigher(org.biojava3.core.sequence.location.template.Point)"><B>isHigher(Point)</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Point.html#isHigher(org.biojava3.core.sequence.location.template.Point)"><B>isHigher(Point)</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</A>
<DD>Returns true if the point is higher in value to the current point
<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#isInitialized()"><B>isInitialized()</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#isLower(org.biojava3.core.sequence.location.template.Point)"><B>isLower(Point)</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Point.html#isLower(org.biojava3.core.sequence.location.template.Point)"><B>isLower(Point)</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</A>
<DD>Returns true if the current point is at a lower position than the
 point given.
<DT><A HREF="../org/biojava3/core/util/SequenceTools.html#isNucleotideSequence(java.lang.String)"><B>isNucleotideSequence(String)</B></A> - 
Static method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html#isStart()"><B>isStart()</B></A> - 
Method in class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)"><B>isStart(AminoAcidCompound)</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</A>
<DD>Returns true if the given compound was a start codon in this
 codon table.
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#isStart()"><B>isStart()</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)"><B>isStart(AminoAcidCompound)</B></A> - 
Method in interface org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A>
<DD>Returns true if the given compound could have been a start amino acid;
 this does not assert if the codon that actually coded for the amino
 acid was a start codon.
<DT><A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html#isStop()"><B>isStop()</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#isUncertain()"><B>isUncertain()</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Point.html#isUncertain()"><B>isUncertain()</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</A>
<DD>Returns a true if the exact point is unknown.
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#isUnknown()"><B>isUnknown()</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Point.html#isUnknown()"><B>isUnknown()</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</A>
<DD>Returns true if the current position is unknown but is
 beyond the position encoded for.
<DT><A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.loader.<A HREF="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</A>
<DD>Iterates through all known sub-locations for this location but does
 not descend
<DT><A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</A>
<DD>Provides basic iterable access to this class
<DT><A HREF="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</A>
<DD>Iterator implementation which attempts to move through the 2D structure
 attempting to skip onto the next sequence as & when it is asked to
<DT><A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.storage.<A HREF="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</A>
<DD>Returns an instance of <CODE>SequenceMixin.SequenceIterator</CODE>
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/views/WindowedSequence.html#iterator()"><B>iterator()</B></A> - 
Method in class org.biojava3.core.sequence.views.<A HREF="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</A>
<DD>Returns an iterator which will return the windows in a sequence in
 sequential order.
<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.html#IUPAC_LOCATION"><B>IUPAC_LOCATION</B></A> - 
Static variable in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io"><B>IUPACParser</B></A> - Class in <A HREF="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A><DD>Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a
 href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c"
 >NCBI</a> with slight modification and put into the classpath resource.<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.html#IUPACParser()"><B>IUPACParser()</B></A> - 
Constructor for class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</A>
<DD>Default version and uses the classpath based IUPAC table
<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.html#IUPACParser(java.io.InputStream)"><B>IUPACParser(InputStream)</B></A> - 
Constructor for class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</A>
<DD>Allows you to specify a different IUPAC table.
<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><B>IUPACParser.IUPACTable</B></A> - Class in <A HREF="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A><DD>Holds the concept of a codon table from the IUPAC format<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#IUPACParser.IUPACTable(java.lang.String, int, java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><B>IUPACParser.IUPACTable(String, int, String, String, String, String, String)</B></A> - 
Constructor for class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#IUPACParser.IUPACTable(java.lang.String, java.lang.Integer, java.lang.String, java.lang.String)"><B>IUPACParser.IUPACTable(String, Integer, String, String)</B></A> - 
Constructor for class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</A>
<DD>Constructor which uses the basic IUPAC codon table format.
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-8.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-10.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-9.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-9.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>

</BODY>
</HTML>
