<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:03 SGT 2011 -->
<TITLE>
R-Index
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="R-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-17.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-19.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-18.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-18.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>
<A NAME="_R_"><!-- --></A><H2>
<B>R</B></H2>
<DL>
<DT><A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#read()"><B>read()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</A>
<DD>Reads a single character.
<DT><A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#read(char[], int, int)"><B>read(char[], int, int)</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</A>
<DD>Reads characters into a portion of an array.
<DT><A HREF="../org/biojava3/core/util/UncompressInputStream.html#read()"><B>read()</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/UncompressInputStream.html#read(byte[], int, int)"><B>read(byte[], int, int)</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.File, boolean)"><B>readFastaDNASequence(File, boolean)</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</A>
<DD>Selecting lazySequenceLoad=true will parse the FASTA file and figure out the accessionid and offsets and return sequence objects
 that can in the future read the sequence from the disk.
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.InputStream)"><B>readFastaDNASequence(InputStream)</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</A>
<DD>Read a fasta DNA sequence
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.File)"><B>readFastaDNASequence(File)</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaProteinSequence(java.io.File)"><B>readFastaProteinSequence(File)</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</A>
<DD>Read a fasta file containing amino acids with setup that would handle most
 cases.
<DT><A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaProteinSequence(java.io.InputStream)"><B>readFastaProteinSequence(InputStream)</B></A> - 
Static method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</A>
<DD>Read a fasta file containing amino acids with setup that would handle most
 cases.
<DT><A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#readLine()"><B>readLine()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</A>
<DD>Reads a line of text.
<DT><A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#ready()"><B>ready()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</A>
<DD>Tells whether this stream is ready to be read.
<DT><A HREF="../org/biojava3/core/sequence/views/WindowViewTests.html#reaminderWindow()"><B>reaminderWindow()</B></A> - 
Method in class org.biojava3.core.sequence.views.<A HREF="../org/biojava3/core/sequence/views/WindowViewTests.html" title="class in org.biojava3.core.sequence.views">WindowViewTests</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#remove()"><B>remove()</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/util/SoftHashMap.html#remove(java.lang.Object)"><B>remove(Object)</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#removeAlignedSequence(S)"><B>removeAlignedSequence(S)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</A>
<DD>Remove a sequence
<DT><A HREF="../org/biojava3/core/sequence/TranscriptSequence.html#removeCDS(java.lang.String)"><B>removeCDS(String)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A>
<DD>Remove a CDS or coding sequence from the transcript sequence
<DT><A HREF="../org/biojava3/core/sequence/GeneSequence.html#removeExon(java.lang.String)"><B>removeExon(String)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>
<DD>Remove the exon sequence
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)"><B>removeFeature(FeatureInterface&lt;AbstractSequence&lt;C&gt;, C&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>Remove a feature from the sequence
<DT><A HREF="../org/biojava3/core/sequence/ChromosomeSequence.html#removeGeneSequence(java.lang.String)"><B>removeGeneSequence(String)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/GeneSequence.html#removeIntron(java.lang.String)"><B>removeIntron(String)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>
<DD>Remove the intron by accession
<DT><A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)"><B>removeNote(String)</B></A> - 
Method in class org.biojava3.core.sequence.template.<A HREF="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/GeneSequence.html#removeTranscript(java.lang.String)"><B>removeTranscript(String)</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>
<DD>Remove the transcript sequence from the gene
<DT><A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#reset()"><B>reset()</B></A> - 
Method in class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</A>
<DD>Resets the stream to the most recent mark.
<DT><A HREF="../org/biojava3/core/util/CRC64Checksum.html#reset()"><B>reset()</B></A> - 
Method in class org.biojava3.core.util.<A HREF="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Point.Resolver.html#resolve(T)"><B>resolve(T)</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Point.Resolver.html" title="interface in org.biojava3.core.sequence.location.template">Point.Resolver</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#respectCase()"><B>respectCase()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#reverse()"><B>reverse()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/FuzzyPoint.html#reverse(int)"><B>reverse(int)</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/SimplePoint.html#reverse(int)"><B>reverse(int)</B></A> - 
Method in class org.biojava3.core.sequence.location.<A HREF="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/location/template/Point.html#reverse(int)"><B>reverse(int)</B></A> - 
Method in interface org.biojava3.core.sequence.location.template.<A HREF="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</A>
<DD>Returns the equivalent position on the reverse strand
<DT><A HREF="../org/biojava3/core/sequence/DNATest.html#reverseComplement()"><B>reverseComplement()</B></A> - 
Method in class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/DNATest.html" title="class in org.biojava3.core.sequence">DNATest</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views"><B>ReversedSequenceView</B></A>&lt;<A HREF="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</A> extends <A HREF="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; - Class in <A HREF="../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A><DD>For a given sequence this class will return the base at the reversed
 position i.e.<DT><A HREF="../org/biojava3/core/sequence/views/ReversedSequenceView.html#ReversedSequenceView(org.biojava3.core.sequence.template.Sequence)"><B>ReversedSequenceView(Sequence&lt;C&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.views.<A HREF="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#rnaAminoAcidTranslator(org.biojava3.core.sequence.transcription.RNAToAminoAcidTranslator)"><B>rnaAminoAcidTranslator(RNAToAminoAcidTranslator)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#rnaCompounds(org.biojava3.core.sequence.template.CompoundSet)"><B>rnaCompounds(CompoundSet&lt;NucleotideCompound&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound"><B>RNACompoundSet</B></A> - Class in <A HREF="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A><DD>&nbsp;<DT><A HREF="../org/biojava3/core/sequence/compound/RNACompoundSet.html#RNACompoundSet()"><B>RNACompoundSet()</B></A> - 
Constructor for class org.biojava3.core.sequence.compound.<A HREF="../org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#rnaCreator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface)"><B>rnaCreator(SequenceCreatorInterface&lt;NucleotideCompound&gt;)</B></A> - 
Method in class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence"><B>RNASequence</B></A> - Class in <A HREF="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A><DD>RNASequence where RNACompoundSet are the allowed values<DT><A HREF="../org/biojava3/core/sequence/RNASequence.html#RNASequence(java.lang.String)"><B>RNASequence(String)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</A>
<DD>Create a RNA sequence from a String
<DT><A HREF="../org/biojava3/core/sequence/RNASequence.html#RNASequence(org.biojava3.core.sequence.template.ProxySequenceReader)"><B>RNASequence(ProxySequenceReader&lt;NucleotideCompound&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</A>
<DD>Create a RNA aequence from a proxy reader
<DT><A HREF="../org/biojava3/core/sequence/RNASequence.html#RNASequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)"><B>RNASequence(String, CompoundSet&lt;NucleotideCompound&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</A>
<DD>Create a RNA sequence from a string with a user defined RNA compound set
<DT><A HREF="../org/biojava3/core/sequence/RNASequence.html#RNASequence(org.biojava3.core.sequence.template.ProxySequenceReader, org.biojava3.core.sequence.template.CompoundSet)"><B>RNASequence(ProxySequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.<A HREF="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</A>
<DD>Create a RNA sequence from a proxy reader and user defined RNA compound set
<DT><A HREF="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io"><B>RNASequenceCreator</B></A> - Class in <A HREF="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A><DD>Used to create a RNA sequence<DT><A HREF="../org/biojava3/core/sequence/io/RNASequenceCreator.html#RNASequenceCreator(org.biojava3.core.sequence.template.CompoundSet)"><B>RNASequenceCreator(CompoundSet&lt;NucleotideCompound&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.io.<A HREF="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views"><B>RnaSequenceView</B></A> - Class in <A HREF="../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A><DD>Attempts to do on the fly translation of RNA by not requesting the compounds
 until asked.<DT><A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html#RnaSequenceView(org.biojava3.core.sequence.template.Sequence)"><B>RnaSequenceView(Sequence&lt;NucleotideCompound&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.views.<A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html#RnaSequenceView(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.CompoundSet)"><B>RnaSequenceView(Sequence&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</B></A> - 
Constructor for class org.biojava3.core.sequence.views.<A HREF="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</A>
<DD>&nbsp;
<DT><A HREF="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription"><B>RNAToAminoAcidTranslator</B></A> - Class in <A HREF="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A><DD>Takes a <A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> of <A HREF="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><CODE>NucleotideCompound</CODE></A> which should
 represent an RNA sequence (<A HREF="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence"><CODE>RNASequence</CODE></A> is good for this) and returns
 a list of <A HREF="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> which hold <A HREF="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><CODE>AminoAcidCompound</CODE></A>.<DT><A HREF="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#RNAToAminoAcidTranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.transcription.Table, boolean, boolean, boolean)"><B>RNAToAminoAcidTranslator(SequenceCreatorInterface&lt;AminoAcidCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;Table.Codon&gt;, CompoundSet&lt;AminoAcidCompound&gt;, Table, boolean, boolean, boolean)</B></A> - 
Constructor for class org.biojava3.core.sequence.transcription.<A HREF="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-17.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-19.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-18.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-18.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">J</A> <A HREF="index-11.html">K</A> <A HREF="index-12.html">L</A> <A HREF="index-13.html">M</A> <A HREF="index-14.html">N</A> <A HREF="index-15.html">O</A> <A HREF="index-16.html">P</A> <A HREF="index-17.html">Q</A> <A HREF="index-18.html">R</A> <A HREF="index-19.html">S</A> <A HREF="index-20.html">T</A> <A HREF="index-21.html">U</A> <A HREF="index-22.html">V</A> <A HREF="index-23.html">W</A> <A HREF="index-24.html">X</A> <HR>

</BODY>
</HTML>
