<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:02 SGT 2011 -->
<TITLE>
FastaWriterHelper
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FastaWriterHelper";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FastaWriterHelper.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterTest.html" title="class in org.biojava3.core.sequence.io"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/io/FastaWriterHelper.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FastaWriterHelper.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.biojava3.core.sequence.io</FONT>
<BR>
Class FastaWriterHelper</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../../resources/inherit.gif" ALT="extended by "><B>org.biojava3.core.sequence.io.FastaWriterHelper</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>FastaWriterHelper</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
The class that should be used to write out fasta file of a sequence collection
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Scooter Willis <willishf at gmail dot com></DD>
</DL>
<HR>

<P>

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#FastaWriterHelper()">FastaWriterHelper</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeGeneSequence(java.io.File, java.util.Collection, boolean)">writeGeneSequence</A></B>(java.io.File&nbsp;file,
                  java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>&gt;&nbsp;geneSequences,
                  boolean&nbsp;showExonUppercase)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeGeneSequence(java.io.OutputStream, java.util.Collection, boolean)">writeGeneSequence</A></B>(java.io.OutputStream&nbsp;outputStream,
                  java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>&gt;&nbsp;geneSequences,
                  boolean&nbsp;showExonUppercase)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeNucleotideSequence(java.io.File, java.util.Collection)">writeNucleotideSequence</A></B>(java.io.File&nbsp;file,
                        java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A>&gt;&nbsp;dnaSequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a collection of NucleotideSequences to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeNucleotideSequence(java.io.OutputStream, java.util.Collection)">writeNucleotideSequence</A></B>(java.io.OutputStream&nbsp;outputStream,
                        java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A>&gt;&nbsp;dnaSequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a collection of NucleotideSequences to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeProteinSequence(java.io.File, java.util.Collection)">writeProteinSequence</A></B>(java.io.File&nbsp;file,
                     java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A>&gt;&nbsp;proteinSequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write collection of protein sequences to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeProteinSequence(java.io.OutputStream, java.util.Collection)">writeProteinSequence</A></B>(java.io.OutputStream&nbsp;outputStream,
                     java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A>&gt;&nbsp;proteinSequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write collection of protein sequences to a stream</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeSequence(java.io.File, org.biojava3.core.sequence.template.Sequence)">writeSequence</A></B>(java.io.File&nbsp;file,
              <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a sequence to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeSequence(java.io.OutputStream, org.biojava3.core.sequence.template.Sequence)">writeSequence</A></B>(java.io.OutputStream&nbsp;outputStream,
              <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a sequence to OutputStream</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeSequences(java.io.OutputStream, java.util.Collection)">writeSequences</A></B>(java.io.OutputStream&nbsp;outputStream,
               java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Method which will write your given Sequences to the specified
 <CODE>OutputStream</CODE>.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="FastaWriterHelper()"><!-- --></A><H3>
FastaWriterHelper</H3>
<PRE>
public <B>FastaWriterHelper</B>()</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="writeProteinSequence(java.io.File, java.util.Collection)"><!-- --></A><H3>
writeProteinSequence</H3>
<PRE>
public static void <B>writeProteinSequence</B>(java.io.File&nbsp;file,
                                        java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A>&gt;&nbsp;proteinSequences)
                                 throws java.lang.Exception</PRE>
<DL>
<DD>Write collection of protein sequences to a file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>file</CODE> - <DD><CODE>proteinSequences</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeProteinSequence(java.io.OutputStream, java.util.Collection)"><!-- --></A><H3>
writeProteinSequence</H3>
<PRE>
public static void <B>writeProteinSequence</B>(java.io.OutputStream&nbsp;outputStream,
                                        java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A>&gt;&nbsp;proteinSequences)
                                 throws java.lang.Exception</PRE>
<DL>
<DD>Write collection of protein sequences to a stream
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outputStream</CODE> - <DD><CODE>proteinSequences</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeGeneSequence(java.io.File, java.util.Collection, boolean)"><!-- --></A><H3>
writeGeneSequence</H3>
<PRE>
public static void <B>writeGeneSequence</B>(java.io.File&nbsp;file,
                                     java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>&gt;&nbsp;geneSequences,
                                     boolean&nbsp;showExonUppercase)
                              throws java.lang.Exception</PRE>
<DL>
<DD>Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>file</CODE> - <DD><CODE>geneSequences</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeGeneSequence(java.io.OutputStream, java.util.Collection, boolean)"><!-- --></A><H3>
writeGeneSequence</H3>
<PRE>
public static void <B>writeGeneSequence</B>(java.io.OutputStream&nbsp;outputStream,
                                     java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A>&gt;&nbsp;geneSequences,
                                     boolean&nbsp;showExonUppercase)
                              throws java.lang.Exception</PRE>
<DL>
<DD>Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outputStream</CODE> - <DD><CODE>dnaSequences</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeNucleotideSequence(java.io.File, java.util.Collection)"><!-- --></A><H3>
writeNucleotideSequence</H3>
<PRE>
public static void <B>writeNucleotideSequence</B>(java.io.File&nbsp;file,
                                           java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A>&gt;&nbsp;dnaSequences)
                                    throws java.lang.Exception</PRE>
<DL>
<DD>Write a collection of NucleotideSequences to a file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>file</CODE> - <DD><CODE>dnaSequences</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeNucleotideSequence(java.io.OutputStream, java.util.Collection)"><!-- --></A><H3>
writeNucleotideSequence</H3>
<PRE>
public static void <B>writeNucleotideSequence</B>(java.io.OutputStream&nbsp;outputStream,
                                           java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A>&gt;&nbsp;dnaSequences)
                                    throws java.lang.Exception</PRE>
<DL>
<DD>Write a collection of NucleotideSequences to a file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outputStream</CODE> - <DD><CODE>dnaSequences</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeSequence(java.io.File, org.biojava3.core.sequence.template.Sequence)"><!-- --></A><H3>
writeSequence</H3>
<PRE>
public static void <B>writeSequence</B>(java.io.File&nbsp;file,
                                 <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&nbsp;sequence)
                          throws java.lang.Exception</PRE>
<DL>
<DD>Write a sequence to a file
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>file</CODE> - <DD><CODE>sequence</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeSequence(java.io.OutputStream, org.biojava3.core.sequence.template.Sequence)"><!-- --></A><H3>
writeSequence</H3>
<PRE>
public static void <B>writeSequence</B>(java.io.OutputStream&nbsp;outputStream,
                                 <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&nbsp;sequence)
                          throws java.lang.Exception</PRE>
<DL>
<DD>Write a sequence to OutputStream
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outputStream</CODE> - <DD><CODE>sequence</CODE> - 
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="writeSequences(java.io.OutputStream, java.util.Collection)"><!-- --></A><H3>
writeSequences</H3>
<PRE>
public static void <B>writeSequences</B>(java.io.OutputStream&nbsp;outputStream,
                                  java.util.Collection&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&gt;&nbsp;sequences)
                           throws java.lang.Exception</PRE>
<DL>
<DD>Method which will write your given Sequences to the specified
 <CODE>OutputStream</CODE>. This is a very generic method which writes just the
 AccessionID of the Sequence as the FASTA header.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outputStream</CODE> - Stream to write to; can be System.out<DD><CODE>sequences</CODE> - The sequences to write out
<DT><B>Throws:</B>
<DD><CODE>java.lang.Exception</CODE> - Thrown normally thanks to IO problems</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FastaWriterHelper.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/FastaWriterTest.html" title="class in org.biojava3.core.sequence.io"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/io/FastaWriterHelper.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FastaWriterHelper.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
