<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:02 SGT 2011 -->
<TITLE>
IUPACParser.IUPACTable
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="IUPACParser.IUPACTable";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/IUPACParser.IUPACTable.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="IUPACParser.IUPACTable.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.biojava3.core.sequence.io</FONT>
<BR>
Class IUPACParser.IUPACTable</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../../resources/inherit.gif" ALT="extended by "><B>org.biojava3.core.sequence.io.IUPACParser.IUPACTable</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A></DD>
</DL>
<DL>
<DT><B>Enclosing class:</B><DD><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public static class <B>IUPACParser.IUPACTable</B><DT>extends java.lang.Object<DT>implements <A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A></DL>
</PRE>

<P>
Holds the concept of a codon table from the IUPAC format
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Andy Yates</DD>
</DL>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
</TABLE>
&nbsp;<A NAME="nested_classes_inherited_from_class_org.biojava3.core.sequence.transcription.Table"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Nested classes/interfaces inherited from interface org.biojava3.core.sequence.transcription.<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</A>, <A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A></CODE></TD>
</TR>
</TABLE>
&nbsp;

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#IUPACParser.IUPACTable(java.lang.String, java.lang.Integer, java.lang.String, java.lang.String)">IUPACParser.IUPACTable</A></B>(java.lang.String&nbsp;name,
                       java.lang.Integer&nbsp;id,
                       java.lang.String&nbsp;aminoAcidString,
                       java.lang.String&nbsp;startCodons)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructor which uses the basic IUPAC codon table format.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#IUPACParser.IUPACTable(java.lang.String, int, java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)">IUPACParser.IUPACTable</A></B>(java.lang.String&nbsp;name,
                       int&nbsp;id,
                       java.lang.String&nbsp;aminoAcidString,
                       java.lang.String&nbsp;startCodons,
                       java.lang.String&nbsp;baseOne,
                       java.lang.String&nbsp;baseTwo,
                       java.lang.String&nbsp;baseThree)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                    <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the compound set of codons</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodons</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucelotides,
          <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a list of codons where the source and target compounds
 are the same as those given by the parameters.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Integer</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getId()">getId</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getName()">getName</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)">isStart</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if the given compound was a start codon in this
 codon table.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="IUPACParser.IUPACTable(java.lang.String, int, java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)"><!-- --></A><H3>
IUPACParser.IUPACTable</H3>
<PRE>
public <B>IUPACParser.IUPACTable</B>(java.lang.String&nbsp;name,
                              int&nbsp;id,
                              java.lang.String&nbsp;aminoAcidString,
                              java.lang.String&nbsp;startCodons,
                              java.lang.String&nbsp;baseOne,
                              java.lang.String&nbsp;baseTwo,
                              java.lang.String&nbsp;baseThree)</PRE>
<DL>
</DL>
<HR>

<A NAME="IUPACParser.IUPACTable(java.lang.String, java.lang.Integer, java.lang.String, java.lang.String)"><!-- --></A><H3>
IUPACParser.IUPACTable</H3>
<PRE>
public <B>IUPACParser.IUPACTable</B>(java.lang.String&nbsp;name,
                              java.lang.Integer&nbsp;id,
                              java.lang.String&nbsp;aminoAcidString,
                              java.lang.String&nbsp;startCodons)</PRE>
<DL>
<DD>Constructor which uses the basic IUPAC codon table format. Useful
 if you need to specify your own IUPAC table with minimal
 definitions from your side.
<P>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getId()"><!-- --></A><H3>
getId</H3>
<PRE>
public java.lang.Integer <B>getId</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getName()"><!-- --></A><H3>
getName</H3>
<PRE>
public java.lang.String <B>getName</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)"><!-- --></A><H3>
isStart</H3>
<PRE>
public boolean <B>isStart</B>(<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&nbsp;compound)
                throws java.lang.IllegalStateException</PRE>
<DL>
<DD>Returns true if the given compound was a start codon in this
 codon table. This will report true if the compound could ever have
 been a start codon.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)">isStart</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.lang.IllegalStateException</CODE> - Thrown if
 <A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)"><CODE>getCodons(CompoundSet, CompoundSet)</CODE></A> was not called first.</DL>
</DD>
</DL>
<HR>

<A NAME="getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
getCodons</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt; <B>getCodons</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucelotides,
                                             <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids)</PRE>
<DL>
<DD>Returns a list of codons where the source and target compounds
 are the same as those given by the parameters.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html#getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodons</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>nucleotides</CODE> - The nucleotide set to use when building BioJava 
 representations of codons<DD><CODE>aminoAcids</CODE> - The target amino acid compounds objects</DL>
</DD>
</DL>
<HR>

<A NAME="getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
getCodonCompoundSet</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt; <B>getCodonCompoundSet</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                                                    <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</PRE>
<DL>
<DD>Returns the compound set of codons
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/IUPACParser.IUPACTable.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="IUPACParser.IUPACTable.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
