<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:03 SGT 2011 -->
<TITLE>
Uses of Interface org.biojava3.core.sequence.template.CompoundSet
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava3.core.sequence.template.CompoundSet";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?org/biojava3/core/sequence/template//class-useCompoundSet.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="CompoundSet.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Interface<br>org.biojava3.core.sequence.template.CompoundSet</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence"><B>org.biojava3.core.sequence</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.compound"><B>org.biojava3.core.sequence.compound</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io"><B>org.biojava3.core.sequence.io</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io.util"><B>org.biojava3.core.sequence.io.util</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.loader"><B>org.biojava3.core.sequence.loader</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.storage"><B>org.biojava3.core.sequence.storage</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.template"><B>org.biojava3.core.sequence.template</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.transcription"><B>org.biojava3.core.sequence.transcription</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.views"><B>org.biojava3.core.sequence.views</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>MultipleSequenceAlignment.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the Compounds defined in the first sequence</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/BasicSequence.html#BasicSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">BasicSequence</A></B>(java.lang.String&nbsp;sequence,
              <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/BasicSequence.html" title="type parameter in BasicSequence">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/ChromosomeSequence.html#ChromosomeSequence(org.biojava3.core.sequence.template.SequenceReader, org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/ChromosomeSequence.html#ChromosomeSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence</A></B>(java.lang.String&nbsp;seqString,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/DNASequence.html#DNASequence(org.biojava3.core.sequence.template.SequenceReader, org.biojava3.core.sequence.template.CompoundSet)">DNASequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader,
            <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a sequence from a ProxySequencereader and user defined compound set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/DNASequence.html#DNASequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">DNASequence</A></B>(java.lang.String&nbsp;seqString,
            <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a sequence from a string with user defined compound set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/ProteinSequence.html#ProteinSequence(org.biojava3.core.sequence.template.ProxySequenceReader, org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;proxyLoader,
                <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A protein sequence where the storage of the sequence is somewhere else with user defined
 set of amino acids.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/ProteinSequence.html#ProteinSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence</A></B>(java.lang.String&nbsp;seqString,
                <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a protein from a string with a user defined set of amino acids</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/RNASequence.html#RNASequence(org.biojava3.core.sequence.template.ProxySequenceReader, org.biojava3.core.sequence.template.CompoundSet)">RNASequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;proxyLoader,
            <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a RNA sequence from a proxy reader and user defined RNA compound set</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/RNASequence.html#RNASequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">RNASequence</A></B>(java.lang.String&nbsp;seqString,
            <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a RNA sequence from a string with a user defined RNA compound set</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.compound"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set of proteinogenic amino acids.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AminoAcidCompound.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html#NucleotideCompound(java.lang.String, org.biojava3.core.sequence.template.CompoundSet, java.lang.String)">NucleotideCompound</A></B>(java.lang.String&nbsp;base,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet,
                   java.lang.String&nbsp;complementStr)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html#NucleotideCompound(java.lang.String, org.biojava3.core.sequence.template.CompoundSet, java.lang.String, org.biojava3.core.sequence.compound.NucleotideCompound[])">NucleotideCompound</A></B>(java.lang.String&nbsp;base,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet,
                   java.lang.String&nbsp;complementStr,
                   <A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>[]&nbsp;constituents)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>IUPACParser.IUPACTable.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the compound set of codons</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>IUPACParser.IUPACTable.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the compound set of codons</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>IUPACParser.IUPACTable.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the compound set of codons</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>IUPACParser.IUPACTable.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodons</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucelotides,
          <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a list of codons where the source and target compounds
 are the same as those given by the parameters.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>IUPACParser.IUPACTable.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodons</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucelotides,
          <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a list of codons where the source and target compounds
 are the same as those given by the parameters.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/DNASequenceCreator.html#DNASequenceCreator(org.biojava3.core.sequence.template.CompoundSet)">DNASequenceCreator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#FileProxyDNASequenceCreator(java.io.File, org.biojava3.core.sequence.template.CompoundSet)">FileProxyDNASequenceCreator</A></B>(java.io.File&nbsp;fastaFile,
                            <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Need File so that we can store full path name in SequenceFileProxyLoader for Random File access as a quick read</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#FileProxyProteinSequenceCreator(java.io.File, org.biojava3.core.sequence.template.CompoundSet)">FileProxyProteinSequenceCreator</A></B>(java.io.File&nbsp;fastaFile,
                                <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Need File so that we can store full path name in SequenceFileProxyLoader for Random File access as a quick read</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#ProteinSequenceCreator(org.biojava3.core.sequence.template.CompoundSet)">ProteinSequenceCreator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/RNASequenceCreator.html#RNASequenceCreator(org.biojava3.core.sequence.template.CompoundSet)">RNASequenceCreator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGType(org.biojava3.core.sequence.template.CompoundSet)">getGCGType</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;C&gt;&nbsp;cs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Determines GCG type</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.loader"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>UniprotProxySequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>StringProxySequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SequenceFileProxyLoader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>UniprotProxySequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>StringProxySequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>SequenceFileProxyLoader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#SequenceFileProxyLoader(java.io.File, org.biojava3.core.sequence.io.template.SequenceParserInterface, long, int, org.biojava3.core.sequence.template.CompoundSet)">SequenceFileProxyLoader</A></B>(java.io.File&nbsp;file,
                        <A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</A>&nbsp;sequenceParser,
                        long&nbsp;sequenceStartIndex,
                        int&nbsp;sequenceLength,
                        <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#StringProxySequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">StringProxySequenceReader</A></B>(java.lang.String&nbsp;sequence,
                          <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#UniprotProxySequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">UniprotProxySequenceReader</A></B>(java.lang.String&nbsp;accession,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The uniprot id is used to retrieve the uniprot XML which is then parsed as a DOM object
 so we know everything about the protein.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.storage"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SingleCompoundSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the compound set given at construction</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>JoiningSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BitSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the compound set backing this store</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BitSequenceReader.BitArrayWorker.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the compound set backing this store</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ArrayListSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SequenceAsStringHelper.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html#getSequenceAsString(java.util.List, org.biojava3.core.sequence.template.CompoundSet, java.lang.Integer, java.lang.Integer, org.biojava3.core.sequence.Strand)">getSequenceAsString</A></B>(java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</A>&gt;&nbsp;parsedCompounds,
                    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</A>&gt;&nbsp;compoundSet,
                    java.lang.Integer&nbsp;bioBegin,
                    java.lang.Integer&nbsp;bioEnd,
                    <A HREF="../../../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>SingleCompoundSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Unsupported</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>JoiningSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BitSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Class is immutable & so this is unsupported</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>ArrayListSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#ArrayListSequenceReader(java.util.List, org.biojava3.core.sequence.template.CompoundSet)">ArrayListSequenceReader</A></B>(java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</A>&gt;&nbsp;compounds,
                        <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#ArrayListSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">ArrayListSequenceReader</A></B>(java.lang.String&nbsp;sequence,
                        <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#BitSequenceReader.BitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int)">BitSequenceReader.BitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</A>&gt;&nbsp;compoundSet,
                                 int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#BitSequenceReader.BitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int[])">BitSequenceReader.BitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</A>&gt;&nbsp;compoundSet,
                                 int[]&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#BitSequenceReader.BitArrayWorker(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">BitSequenceReader.BitArrayWorker</A></B>(java.lang.String&nbsp;sequence,
                                 <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#FourBitSequenceReader.FourBitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int)">FourBitSequenceReader.FourBitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</A>&gt;&nbsp;compoundSet,
                                         int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#FourBitSequenceReader.FourBitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int[])">FourBitSequenceReader.FourBitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</A>&gt;&nbsp;compoundSet,
                                         int[]&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#FourBitSequenceReader.FourBitArrayWorker(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">FourBitSequenceReader.FourBitArrayWorker</A></B>(java.lang.String&nbsp;sequence,
                                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#FourBitSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">FourBitSequenceReader</A></B>(java.lang.String&nbsp;sequence,
                      <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#FourBitSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.AccessionID)">FourBitSequenceReader</A></B>(java.lang.String&nbsp;sequence,
                      <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</A>&gt;&nbsp;compoundSet,
                      <A HREF="../../../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava3.core.sequence.template.CompoundSet, java.util.List)">JoiningSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&nbsp;compoundSet,
                      java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.Sequence...)">JoiningSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&nbsp;compoundSet,
                      <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;...&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#SingleCompoundSequenceReader(C, org.biojava3.core.sequence.template.CompoundSet, int)">SingleCompoundSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</A>&nbsp;compound,
                             <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</A>&gt;&nbsp;compoundSet,
                             int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Build the object with a compound rather than a String</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#SingleCompoundSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet, int)">SingleCompoundSequenceReader</A></B>(java.lang.String&nbsp;compound,
                             <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</A>&gt;&nbsp;compoundSet,
                             int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Public constructor to be used with String based constructor</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int)">TwoBitSequenceReader.TwoBitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</A>&gt;&nbsp;compoundSet,
                                       int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(org.biojava3.core.sequence.template.CompoundSet, int[])">TwoBitSequenceReader.TwoBitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</A>&gt;&nbsp;compoundSet,
                                       int[]&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">TwoBitSequenceReader.TwoBitArrayWorker</A></B>(java.lang.String&nbsp;sequence,
                                       <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">TwoBitSequenceReader</A></B>(java.lang.String&nbsp;sequence,
                     <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(java.lang.String, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.AccessionID)">TwoBitSequenceReader</A></B>(java.lang.String&nbsp;sequence,
                     <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</A>&gt;&nbsp;compoundSet,
                     <A HREF="../../../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.template"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet&lt;C extends NucleotideCompound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SequenceProxyView.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Sequence.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Gets the compound set used to back this Sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>LightweightProfile.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/LightweightProfile.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><CODE>CompoundSet</CODE></A> of all <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>s</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AbstractSequence.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AbstractCompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getFromCompoundSet()">getFromCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AbstractCompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getToCompoundSet()">getToCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>SequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="type parameter in SequenceReader">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AbstractSequence.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#AbstractCompoundTranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">AbstractCompoundTranslator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;&nbsp;creator,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</A>&gt;&nbsp;fromCompoundSet,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;&nbsp;toCompoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#AbstractCompoundTranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">AbstractCompoundTranslator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;&nbsp;creator,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</A>&gt;&nbsp;fromCompoundSet,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;&nbsp;toCompoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#AbstractSequence(org.biojava3.core.sequence.template.SequenceReader, org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;proxyLoader,
                 <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A ProxySequenceReader allows abstraction of both the storage of the sequence data and the location
 of the sequence data.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#AbstractSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</A></B>(java.lang.String&nbsp;seqString,
                 <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a Sequence from a simple string where the values should be found in compoundSet</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.transcription"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getAminoAcidCompounds()">getAminoAcidCompounds</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Table.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDnaCompounds()">getDnaCompounds</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaCompounds()">getRnaCompounds</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A></CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.Builder.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#aminoAcidsCompounds(org.biojava3.core.sequence.template.CompoundSet)">aminoAcidsCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A></CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.Builder.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#dnaCompounds(org.biojava3.core.sequence.template.CompoundSet)">dnaCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Table.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Table.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds,
                    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcidCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Table.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html#getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodons</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucelotides,
          <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Table.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html#getCodons(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet)">getCodons</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucelotides,
          <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</A></CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.Builder.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#rnaCompounds(org.biojava3.core.sequence.template.CompoundSet)">rnaCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#DNAToRNATranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, boolean)">DNAToRNATranslator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCreator,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rna,
                   boolean&nbsp;shortCutTranslation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#DNAToRNATranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, boolean)">DNAToRNATranslator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCreator,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna,
                   <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rna,
                   boolean&nbsp;shortCutTranslation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#RNAToAminoAcidTranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.transcription.Table, boolean, boolean, boolean)">RNAToAminoAcidTranslator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;creator,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucleotides,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;&nbsp;codons,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids,
                         <A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A>&nbsp;table,
                         boolean&nbsp;trimStops,
                         boolean&nbsp;initMetOnly,
                         boolean&nbsp;translateNCodons)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#RNAToAminoAcidTranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.transcription.Table, boolean, boolean, boolean)">RNAToAminoAcidTranslator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;creator,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucleotides,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;&nbsp;codons,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids,
                         <A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A>&nbsp;table,
                         boolean&nbsp;trimStops,
                         boolean&nbsp;initMetOnly,
                         boolean&nbsp;translateNCodons)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#RNAToAminoAcidTranslator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.transcription.Table, boolean, boolean, boolean)">RNAToAminoAcidTranslator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;creator,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;nucleotides,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A>&gt;&nbsp;codons,
                         <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;aminoAcids,
                         <A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</A>&nbsp;table,
                         boolean&nbsp;trimStops,
                         boolean&nbsp;initMetOnly,
                         boolean&nbsp;translateNCodons)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.views"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A> in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>RnaSequenceView.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/views/RnaSequenceView.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/RnaSequenceView.html#RnaSequenceView(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.CompoundSet)">RnaSequenceView</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;sourceDna,
                <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?org/biojava3/core/sequence/template//class-useCompoundSet.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="CompoundSet.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
