<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:03 SGT 2011 -->
<TITLE>
Uses of Interface org.biojava3.core.sequence.template.Compound
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava3.core.sequence.template.Compound";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?org/biojava3/core/sequence/template//class-useCompound.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Compound.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Interface<br>org.biojava3.core.sequence.template.Compound</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence"><B>org.biojava3.core.sequence</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.compound"><B>org.biojava3.core.sequence.compound</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.edits"><B>org.biojava3.core.sequence.edits</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.features"><B>org.biojava3.core.sequence.features</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io"><B>org.biojava3.core.sequence.io</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io.template"><B>org.biojava3.core.sequence.io.template</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io.util"><B>org.biojava3.core.sequence.io.util</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.loader"><B>org.biojava3.core.sequence.loader</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.location"><B>org.biojava3.core.sequence.location</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.location.template"><B>org.biojava3.core.sequence.location.template</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.storage"><B>org.biojava3.core.sequence.storage</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.template"><B>org.biojava3.core.sequence.template</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.transcription"><B>org.biojava3.core.sequence.transcription</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.views"><B>org.biojava3.core.sequence.views</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Bare bones version of the Sequence object to be used sparingly.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment&lt;S extends Sequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements a minimal data structure for reading and writing a sequence alignment.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.compound"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used to describe an Amino Acid.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Define a codon</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>NucleotideCompound.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>CodonCompound.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/CodonCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AminoAcidCompound.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.edits"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.html" title="interface in org.biojava3.core.sequence.edits">Edit&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Interface for carrying out edit operations on a Sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Abstract class which defines all edit operations as a call to discover
 what 5' and 3' ends of an editing Sequence should be joined together
 with a target Sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implementation which allows for the deletion of bases from a Sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Edit implementation which allows us to insert a base at any position
 in a Sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Allows for the substitution of bases into an existing Sequence.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.features"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature&lt;S extends AbstractSequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface&lt;S extends AbstractSequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Interface class to handle describing arbitrary features.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature&lt;S extends AbstractSequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;It is common to have a numerical value or values associated with a feature.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/features/TextFeature.html" title="class in org.biojava3.core.sequence.features">TextFeature&lt;S extends AbstractSequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A implmentation of AbstractFeature</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the
 primary class used to read Fasta files</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The FastaWriter writes a collection of sequences to an outputStream.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat&lt;S extends AbstractSequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser&lt;S extends AbstractSequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface

 GenBank                           gi|gi-number|gb|accession|locus
 ENA Data Library                 gi|gi-number|emb|accession|locus
 DDBJ, DNA Database of Japan       gi|gi-number|dbj|accession|locus
 NBRF PIR                          pir||entry
 Protein Research Foundation       prf||name
 SWISS-PROT                        sp|accession|name
 Brookhaven Protein Data Bank (1)  pdb|entry|chain
 Brookhaven Protein Data Bank (2)  entry:chain|PDBID|CHAIN|SEQUENCE
 PDB EBI                           PDB:1ECY_A mol:protein length:142  ECOTIN
 Patents                           pat|country|number
 GenInfo Backbone Id               bbs|number
 General database identifier       gnl|database|identifier
 NCBI Reference Sequence           ref|accession|locus
 Local Sequence identifier         lcl|identifier</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io.template"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(java.util.List)">getGCGChecksum</A></B>(java.util.List&lt;S&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates GCG checksum for entire list of sequences</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(S)">getGCGChecksum</A></B>(S&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates GCG checksum for a given sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGHeader(java.util.List)">getGCGHeader</A></B>(java.util.List&lt;S&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Assembles a GCG file header</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGType(org.biojava3.core.sequence.template.CompoundSet)">getGCGType</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;C&gt;&nbsp;cs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Determines GCG type</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getIDFormat(java.util.List)">getIDFormat</A></B>(java.util.List&lt;S&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Creates format String for accession IDs</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.loader"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">ArrayListProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An example of a ProxySequenceReader that is created from a String.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the sequence data and other data elements
 associated with the ProteinSequence by Uniprot.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>UniprotProxySequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>StringProxySequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SequenceFileProxyLoader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.location"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation&lt;S extends AbstractSequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A location in a sequence that keeps a reference to its parent sequence</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.location.template"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Location.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/Location.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will return a SequenceReader object which offers a view of all resolved
 locations i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>AbstractLocation.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Location.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/Location.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will return a SequenceReader object which represents the outer bounds
 of this Location</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>AbstractLocation.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/AbstractLocation.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;If circular this will return the sequence represented by the sub
 locations joined.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</A> that return types with arguments of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</A>&lt;? extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AccesionedLocation.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/AccesionedLocation.html#getProxySequenceReader()">getProxySequenceReader</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Return the proxy reader used to get sequence for this location.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.storage"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores a Sequence as a collection of compounds in an ArrayList</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An implementation of the popular bit encodings.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The logic of working with a bit has been separated out into this class
 to help developers create the bit data structures without having to
 put the code into an intermediate format and to also use the format
 without the need to copy this code.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Four bit encoding of the bit formats.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A four bit per compound implementation of the bit array worker code.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This reader actually proxies onto multiple types of sequence in order
 to allow a number of sequence objects to act as if they are one sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is a common method that can be used across multiple storage/proxy implementations to
 handle Negative strand and other interesting elements of sequence data.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SingleCompoundSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Delegates to <CODE>SequenceMixin#countCompounds(org.biojava3.core.sequence.template.Sequence, C[]) </CODE></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>JoiningSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>BitSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Counts the number of times a compound appears in this sequence store</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>ArrayListSequenceReader.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.template"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator&lt;F extends Compound,T extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator&lt;F extends Compound,T extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The base class for DNA, RNA and Protein sequences.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator&lt;F extends Compound,T extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator&lt;F extends Compound,T extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile&lt;S extends Sequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Defines a minimal data structure for reading and writing a sequence alignment.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Main interface for defining a collection of Compounds and accessing them
 using biological indexes</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A basic sequence iterator which iterates over the given Sequence by
 biological index.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Subinterfaces of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The details of a Compound</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#checksum(org.biojava3.core.sequence.template.Sequence)">checksum</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a simple CRC64 checksum on any given sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#countCompounds(org.biojava3.core.sequence.template.Sequence, C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
               C...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For the given vargs of compounds this method counts the number of
 times those compounds appear in the given sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.Iterator&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#createIterator(org.biojava3.core.sequence.template.Sequence)">createIterator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Creates a simple sequence iterator which moves through a sequence going
 from 1 to the length of the Sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#createSubSequence(org.biojava3.core.sequence.template.Sequence, int, int)">createSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
                  int&nbsp;start,
                  int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Creates a simple sub sequence view delimited by the given start and end.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.Map&lt;C,java.lang.Integer&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)">getComposition</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Does a linear scan over the given Sequence and records the number of
 times each base appears.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.Map&lt;C,java.lang.Double&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#getDistribution(org.biojava3.core.sequence.template.Sequence)">getDistribution</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Analogous to <A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)"><CODE>SequenceMixin.getComposition(Sequence)</CODE></A> but returns the
 distribution of that <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><CODE>Compound</CODE></A> over the given sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence, C)">indexOf</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
        C&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a linear search of the given Sequence for the given compound.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#inverse(org.biojava3.core.sequence.template.Sequence)">inverse</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A method which attempts to do the right thing when is comes to a
 reverse/reverse complement</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#lastIndexOf(org.biojava3.core.sequence.template.Sequence, C)">lastIndexOf</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
            C&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a reversed linear search of the given Sequence by wrapping
 it in a <A HREF="../../../../../../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views"><CODE>ReversedSequenceView</CODE></A> and passing it into
 <A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence, C)"><CODE>SequenceMixin.indexOf(Sequence, Compound)</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#nonOverlappingKmers(org.biojava3.core.sequence.template.Sequence, int)">nonOverlappingKmers</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
                    int&nbsp;kmer)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Produces kmers of the specified size e.g.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#overlappingKmers(org.biojava3.core.sequence.template.Sequence, int)">overlappingKmers</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
                 int&nbsp;kmer)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used to generate overlapping k-mers such i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
boolean</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)">sequenceEquality</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;source,
                 <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A case-sensitive manner of comparing two sequence objects together.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
boolean</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;source,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A case-insensitive manner of comparing two sequence objects together.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#shuffle(org.biojava3.core.sequence.template.Sequence)">shuffle</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements sequence shuffling by first materializing the given
 <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> into a <CODE>List</CODE>, applying
 <CODE>Collections.shuffle(List)</CODE> and then returning the shuffled
 elements in a new instance of <CODE>SequenceBackingStore</CODE> which behaves
 as a <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.List&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)">toList</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For the given <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> this will return a <CODE>List</CODE> filled with
 the Compounds of that <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toString(org.biojava3.core.sequence.template.Sequence)">toString</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Shortcut to <A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)"><CODE>SequenceMixin.toStringBuilder(org.biojava3.core.sequence.template.Sequence)</CODE></A>
 which calls toString() on the resulting object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.StringBuilder</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)">toStringBuilder</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For the given Sequence this will return a <CODE>StringBuilder</CODE> object
 filled with the results of <CODE>Compound#toString()</CODE>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
void</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#write(java.lang.Appendable, org.biojava3.core.sequence.template.Sequence)">write</A></B>(java.lang.Appendable&nbsp;appendable,
      <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used as a way of sending a Sequence to a writer without the cost of
 converting to a full length String and then writing the data out</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SequenceProxyView.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>Sequence.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the number of times we found a compound in the Sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>AbstractSequence.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#countCompounds(org.biojava3.core.sequence.template.Sequence, C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
               C...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For the given vargs of compounds this method counts the number of
 times those compounds appear in the given sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Compound.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AbstractCompound.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.transcription"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Attempts to wrap compounds so it is possible to view them
 in a case insensitive manner</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Instance of a Codon which is 3 <A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><CODE>NucleotideCompound</CODE></A>s, its
 corresponding <A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><CODE>AminoAcidCompound</CODE></A> and if it is a start or stop codon.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Table.Codon.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Table.Codon.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>CaseInsensitiveCompound.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.views"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A> in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For a given sequence this class will return the base at the reversed
 position i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A sliding window view of a sequence which does not implement any
 interfaces like <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> because they do not fit how this works.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?org/biojava3/core/sequence/template//class-useCompound.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Compound.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
