<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:03 SGT 2011 -->
<TITLE>
Uses of Interface org.biojava3.core.sequence.template.Sequence
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava3.core.sequence.template.Sequence";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?org/biojava3/core/sequence/template//class-useSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Sequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Interface<br>org.biojava3.core.sequence.template.Sequence</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence"><B>org.biojava3.core.sequence</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.compound"><B>org.biojava3.core.sequence.compound</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.edits"><B>org.biojava3.core.sequence.edits</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io"><B>org.biojava3.core.sequence.io</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io.template"><B>org.biojava3.core.sequence.io.template</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.io.util"><B>org.biojava3.core.sequence.io.util</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.loader"><B>org.biojava3.core.sequence.loader</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.location.template"><B>org.biojava3.core.sequence.location.template</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.storage"><B>org.biojava3.core.sequence.storage</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.template"><B>org.biojava3.core.sequence.template</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.transcription"><B>org.biojava3.core.sequence.transcription</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.sequence.views"><B>org.biojava3.core.sequence.views</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.biojava3.core.util"><B>org.biojava3.core.util</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment&lt;S extends Sequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements a minimal data structure for reading and writing a sequence alignment.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Bare bones version of the Sequence object to be used sparingly.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Represents a exon or coding sequence in a gene.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A ChromosomeSequence is a DNASequence but keeps track of geneSequences</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is class should model the attributes associated with a DNA sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A gene contains a collection of Exon sequences</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The representation of a ProteinSequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;RNASequence where RNACompoundSet are the allowed values</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used to map the start codon feature on a gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used to map the stop codon sequence on a gene</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This is the sequence if you want to go from a gene sequence to a protein sequence.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.compound"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AminoAcidCompoundSet.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#verifySequence(org.biojava3.core.sequence.template.Sequence)">verifySequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.edits"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.html" title="type parameter in Edit">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Edit.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.html#edit(org.biojava3.core.sequence.template.Sequence)">edit</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.html" title="type parameter in Edit">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Edit.AbstractEdit.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#edit(org.biojava3.core.sequence.template.Sequence)">edit</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;&nbsp;editingSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Edit.AbstractEdit.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava3.core.sequence.template.Sequence)">getTargetSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;&nbsp;editingSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the Sequence which is our edit.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.html" title="type parameter in Edit">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Edit.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.html#edit(org.biojava3.core.sequence.template.Sequence)">edit</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.html" title="type parameter in Edit">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Edit.AbstractEdit.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#edit(org.biojava3.core.sequence.template.Sequence)">edit</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;&nbsp;editingSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>Edit.Substitute.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Substitute.html#getEnd(org.biojava3.core.sequence.template.Sequence)">getEnd</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Must use this rather than the no-args getEnd as this can return
 -1 and the length of a sub is dependent on the length of the
 Sequence; we cannot assume 1:1 mapping between characters in a
 String and the number of compounds we will have to insert.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Edit.AbstractEdit.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava3.core.sequence.template.Sequence)">getTargetSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</A>&gt;&nbsp;editingSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the Sequence which is our edit.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Insert.html#Edit.Insert(org.biojava3.core.sequence.template.Sequence, int)">Edit.Insert</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</A>&gt;&nbsp;sequence,
            int&nbsp;position)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Insert.html#Edit.Insert(org.biojava3.core.sequence.template.Sequence, int, int)">Edit.Insert</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</A>&gt;&nbsp;sequence,
            int&nbsp;start,
            int&nbsp;stop)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Substitute.html#Edit.Substitute(org.biojava3.core.sequence.template.Sequence, int)">Edit.Substitute</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</A>&gt;&nbsp;sequence,
                int&nbsp;position)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the
 primary class used to read Fasta files</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The FastaWriter writes a collection of sequences to an outputStream.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>FastaWriterHelper.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeSequence(java.io.File, org.biojava3.core.sequence.template.Sequence)">writeSequence</A></B>(java.io.File&nbsp;file,
              <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a sequence to a file</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>FastaWriterHelper.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeSequence(java.io.OutputStream, org.biojava3.core.sequence.template.Sequence)">writeSequence</A></B>(java.io.OutputStream&nbsp;outputStream,
              <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Write a sequence to OutputStream</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../../../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</A> with type arguments of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>FastaWriterHelper.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/FastaWriterHelper.html#writeSequences(java.io.OutputStream, java.util.Collection)">writeSequences</A></B>(java.io.OutputStream&nbsp;outputStream,
               java.util.Collection&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Method which will write your given Sequences to the specified
 <CODE>OutputStream</CODE>.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io.template"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface&lt;S extends Sequence&lt;?&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.io.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(java.util.List)">getGCGChecksum</A></B>(java.util.List&lt;S&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates GCG checksum for entire list of sequences</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(S)">getGCGChecksum</A></B>(S&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates GCG checksum for a given sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGHeader(java.util.List)">getGCGHeader</A></B>(java.util.List&lt;S&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Assembles a GCG file header</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;S extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;,C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>IOUtils.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/io/util/IOUtils.html#getIDFormat(java.util.List)">getIDFormat</A></B>(java.util.List&lt;S&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Creates format String for accession IDs</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.loader"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">ArrayListProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An example of a ProxySequenceReader that is created from a String.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the sequence data and other data elements
 associated with the ProteinSequence by Uniprot.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.location.template"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Location.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/Location.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will return a SequenceReader object which offers a view of all resolved
 locations i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>AbstractLocation.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Location.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/Location.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will return a SequenceReader object which represents the outer bounds
 of this Location</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>AbstractLocation.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/AbstractLocation.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;If circular this will return the sequence represented by the sub
 locations joined.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Location.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/Location.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will return a SequenceReader object which offers a view of all resolved
 locations i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>AbstractLocation.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Location.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/Location.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will return a SequenceReader object which represents the outer bounds
 of this Location</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>AbstractLocation.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/location/template/AbstractLocation.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;If circular this will return the sequence represented by the sub
 locations joined.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.storage"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores a Sequence as a collection of compounds in an ArrayList</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An implementation of the popular bit encodings.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Four bit encoding of the bit formats.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This reader actually proxies onto multiple types of sequence in order
 to allow a number of sequence objects to act as if they are one sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader&lt;C extends NucleotideCompound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implementation of the 2bit encoding.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BitSequenceReader.BitArrayWorker.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#populate(org.biojava3.core.sequence.template.Sequence)">populate</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Loops through the Compounds in a Sequence and passes them onto
 <A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C, int)"><CODE>BitSequenceReader.BitArrayWorker.setCompoundAt(Compound, int)</CODE></A></TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#BitSequenceReader.BitArrayWorker(org.biojava3.core.sequence.template.Sequence)">BitSequenceReader.BitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#FourBitSequenceReader.FourBitArrayWorker(org.biojava3.core.sequence.template.Sequence)">FourBitSequenceReader.FourBitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#FourBitSequenceReader(org.biojava3.core.sequence.template.Sequence)">FourBitSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava3.core.sequence.template.CompoundSet, org.biojava3.core.sequence.template.Sequence...)">JoiningSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&nbsp;compoundSet,
                      <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;...&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava3.core.sequence.template.Sequence...)">JoiningSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;...&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Allows creation of the store from Vargs Sequence<C> objects.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(org.biojava3.core.sequence.template.Sequence)">TwoBitSequenceReader.TwoBitArrayWorker</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(org.biojava3.core.sequence.template.Sequence)">TwoBitSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../../../../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</A> with type arguments of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava3.core.sequence.template.CompoundSet, java.util.List)">JoiningSequenceReader</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&nbsp;compoundSet,
                      java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(java.util.List)">JoiningSequenceReader</A></B>(java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</A>&gt;&gt;&nbsp;sequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Allows creation of the store from List<Sequence<C>>.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.template"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with type parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile&lt;S extends Sequence&lt;C&gt;,C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Defines a minimal data structure for reading and writing a sequence alignment.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Subinterfaces of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;interface</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The base class for DNA, RNA and Protein sequences.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>CompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AbstractCompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SequenceView.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html#getViewedSequence()">getViewedSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SequenceProxyView.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html#getViewedSequence()">getViewedSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#shuffle(org.biojava3.core.sequence.template.Sequence)">shuffle</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements sequence shuffling by first materializing the given
 <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> into a <CODE>List</CODE>, applying
 <CODE>Collections.shuffle(List)</CODE> and then returning the shuffled
 elements in a new instance of <CODE>SequenceBackingStore</CODE> which behaves
 as a <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> that return types with arguments of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>CompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>AbstractCompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#checksum(org.biojava3.core.sequence.template.Sequence)">checksum</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a simple CRC64 checksum on any given sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#countAT(org.biojava3.core.sequence.template.Sequence)">countAT</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the count of AT in the given sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#countCompounds(org.biojava3.core.sequence.template.Sequence, C...)">countCompounds</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
               C...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For the given vargs of compounds this method counts the number of
 times those compounds appear in the given sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#countGC(org.biojava3.core.sequence.template.Sequence)">countGC</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the count of GC in the given sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.Iterator&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#createIterator(org.biojava3.core.sequence.template.Sequence)">createIterator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Creates a simple sequence iterator which moves through a sequence going
 from 1 to the length of the Sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>CompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>AbstractCompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>CompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>AbstractCompoundTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#createSubSequence(org.biojava3.core.sequence.template.Sequence, int, int)">createSubSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
                  int&nbsp;start,
                  int&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Creates a simple sub sequence view delimited by the given start and end.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.Map&lt;C,java.lang.Integer&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)">getComposition</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Does a linear scan over the given Sequence and records the number of
 times each base appears.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.Map&lt;C,java.lang.Double&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#getDistribution(org.biojava3.core.sequence.template.Sequence)">getDistribution</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Analogous to <A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)"><CODE>SequenceMixin.getComposition(Sequence)</CODE></A> but returns the
 distribution of that <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><CODE>Compound</CODE></A> over the given sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence, C)">indexOf</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
        C&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a linear search of the given Sequence for the given compound.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#inverse(org.biojava3.core.sequence.template.Sequence)">inverse</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A method which attempts to do the right thing when is comes to a
 reverse/reverse complement</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
int</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#lastIndexOf(org.biojava3.core.sequence.template.Sequence, C)">lastIndexOf</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
            C&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a reversed linear search of the given Sequence by wrapping
 it in a <A HREF="../../../../../../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views"><CODE>ReversedSequenceView</CODE></A> and passing it into
 <A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence, C)"><CODE>SequenceMixin.indexOf(Sequence, Compound)</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#nonOverlappingKmers(org.biojava3.core.sequence.template.Sequence, int)">nonOverlappingKmers</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
                    int&nbsp;kmer)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Produces kmers of the specified size e.g.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;C&gt;&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#overlappingKmers(org.biojava3.core.sequence.template.Sequence, int)">overlappingKmers</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence,
                 int&nbsp;kmer)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used to generate overlapping k-mers such i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
boolean</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)">sequenceEquality</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;source,
                 <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A case-sensitive manner of comparing two sequence objects together.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
boolean</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)">sequenceEquality</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;source,
                 <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A case-sensitive manner of comparing two sequence objects together.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
boolean</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;source,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A case-insensitive manner of comparing two sequence objects together.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
boolean</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;source,
                           <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A case-insensitive manner of comparing two sequence objects together.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#shuffle(org.biojava3.core.sequence.template.Sequence)">shuffle</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Implements sequence shuffling by first materializing the given
 <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> into a <CODE>List</CODE>, applying
 <CODE>Collections.shuffle(List)</CODE> and then returning the shuffled
 elements in a new instance of <CODE>SequenceBackingStore</CODE> which behaves
 as a <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.util.List&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)">toList</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For the given <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A> this will return a <CODE>List</CODE> filled with
 the Compounds of that <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><CODE>Sequence</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.String</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toString(org.biojava3.core.sequence.template.Sequence)">toString</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Shortcut to <A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)"><CODE>SequenceMixin.toStringBuilder(org.biojava3.core.sequence.template.Sequence)</CODE></A>
 which calls toString() on the resulting object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
java.lang.StringBuilder</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)">toStringBuilder</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For the given Sequence this will return a <CODE>StringBuilder</CODE> object
 filled with the results of <CODE>Compound#toString()</CODE>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>CompoundSet.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html#verifySequence(org.biojava3.core.sequence.template.Sequence)">verifySequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="type parameter in CompoundSet">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AbstractCompoundSet.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundSet.html#verifySequence(org.biojava3.core.sequence.template.Sequence)">verifySequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt; 
<BR>
void</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>SequenceMixin.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.html#write(java.lang.Appendable, org.biojava3.core.sequence.template.Sequence)">write</A></B>(java.lang.Appendable&nbsp;appendable,
      <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used as a way of sending a Sequence to a writer without the cost of
 converting to a full length String and then writing the data out</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#SequenceMixin.SequenceIterator(org.biojava3.core.sequence.template.Sequence)">SequenceMixin.SequenceIterator</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html#SequenceProxyView(org.biojava3.core.sequence.template.Sequence)">SequenceProxyView</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html#SequenceProxyView(org.biojava3.core.sequence.template.Sequence, java.lang.Integer, java.lang.Integer)">SequenceProxyView</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</A>&gt;&nbsp;sequence,
                  java.lang.Integer&nbsp;bioStart,
                  java.lang.Integer&nbsp;bioEnd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Main constructor for working with SequenceProxyViews</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.transcription"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>DNAToRNATranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>DNAToRNATranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.transcription.Frame)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence,
               <A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>&nbsp;frame)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#translate(org.biojava3.core.sequence.template.Sequence)">translate</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Quick method to let you go from a CDS to a Peptide quickly.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Frame.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html#wrap(org.biojava3.core.sequence.template.Sequence)">wrap</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;incoming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Optionally wraps a Sequence in a reverse complementing view (if the
 frame is on the reverse strand) and creates a sub sequence view if
 it is required.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> that return types with arguments of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>RNAToAminoAcidTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs the core conversion of RNA to Peptide.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>DNAToRNATranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Overloaded local version which delegates to an optional translator
 when told to (specified during construction).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>,<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.transcription.Frame...)">multipleFrameTranslation</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna,
                         <A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>...&nbsp;frames)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A way of translating DNA in a number of frames</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>DNAToRNATranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>DNAToRNATranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.transcription.Frame)">createSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence,
               <A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>&nbsp;frame)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>RNAToAminoAcidTranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs the core conversion of RNA to Peptide.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>DNAToRNATranslator.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;originalSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Overloaded local version which delegates to an optional translator
 when told to (specified during construction).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;<A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>,<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.transcription.Frame...)">multipleFrameTranslation</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna,
                         <A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</A>...&nbsp;frames)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A way of translating DNA in a number of frames</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>TranscriptionEngine.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#translate(org.biojava3.core.sequence.template.Sequence)">translate</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;dna)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Quick method to let you go from a CDS to a Peptide quickly.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="0" SUMMARY="">
<TR ALIGN="right" VALIGN="">
<TD NOWRAP><FONT SIZE="-1">
<CODE>&lt;C extends <A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt; 
<BR>
<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</CODE></FONT></TD>
</TR>
</TABLE>
</CODE></FONT></TD>
<TD><CODE><B>Frame.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/transcription/Frame.html#wrap(org.biojava3.core.sequence.template.Sequence)">wrap</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;&nbsp;incoming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Optionally wraps a Sequence in a reverse complementing view (if the
 frame is on the reverse strand) and creates a sub sequence view if
 it is required.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.sequence.views"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Classes in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A> that implement <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView&lt;C extends ComplementCompound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For a given sequence this class will create a view over the top of it
 and for every request the code will return the complement of the underlying
 base e.g.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView&lt;C extends Compound&gt;</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For a given sequence this class will return the base at the reversed
 position i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Attempts to do on the fly translation of RNA by not requesting the compounds
 until asked.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>WindowedSequence.</B><B><A HREF="../../../../../../org/biojava3/core/sequence/views/WindowedSequence.html#getBackingSequence()">getBackingSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Access the sequence which backs this window</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../../../../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</A> with parameters of type <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/ComplementSequenceView.html#ComplementSequenceView(org.biojava3.core.sequence.template.Sequence)">ComplementSequenceView</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/ReversedSequenceView.html#ReversedSequenceView(org.biojava3.core.sequence.template.Sequence)">ReversedSequenceView</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</A>&gt;&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/RnaSequenceView.html#RnaSequenceView(org.biojava3.core.sequence.template.Sequence)">RnaSequenceView</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;sourceDna)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/RnaSequenceView.html#RnaSequenceView(org.biojava3.core.sequence.template.Sequence, org.biojava3.core.sequence.template.CompoundSet)">RnaSequenceView</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;sourceDna,
                <A HREF="../../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</A>&gt;&nbsp;rnaCompounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../org/biojava3/core/sequence/views/WindowedSequence.html#WindowedSequence(org.biojava3.core.sequence.template.Sequence, int)">WindowedSequence</A></B>(<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../../org/biojava3/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</A>&gt;&nbsp;sequence,
                 int&nbsp;windowSize)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.biojava3.core.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A> in <A HREF="../../../../../../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</A> that return <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;?&gt;</CODE></FONT></TD>
<TD><CODE><B>SequenceTools.</B><B><A HREF="../../../../../../org/biojava3/core/util/SequenceTools.html#getSeqeunceFromString(java.lang.String)">getSeqeunceFromString</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?org/biojava3/core/sequence/template//class-useSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Sequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
