<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Tue May 24 20:30:02 SGT 2011 -->
<TITLE>
AbstractSequence
</TITLE>

<META NAME="date" CONTENT="2011-05-24">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AbstractSequence";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/AbstractSequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/template/AbstractSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="AbstractSequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.biojava3.core.sequence.template</FONT>
<BR>
Class AbstractSequence&lt;C extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../../../resources/inherit.gif" ALT="extended by "><B>org.biojava3.core.sequence.template.AbstractSequence&lt;C&gt;</B>
</PRE>
<DL>
<DT><DT><B>Type Parameters:</B><DD><CODE>C</CODE> - </DL>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.lang.Iterable&lt;C&gt;, <A HREF="../../../../../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</A>, <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</DD>
</DL>
<DL>
<DT><B>Direct Known Subclasses:</B> <DD><A HREF="../../../../../org/biojava3/core/sequence/BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence</A>, <A HREF="../../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</A>, <A HREF="../../../../../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</A>, <A HREF="../../../../../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public abstract class <B>AbstractSequence&lt;C extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</B><DT>extends java.lang.Object<DT>implements <A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;C&gt;</DL>
</PRE>

<P>
The base class for DNA, RNA and Protein sequences.
<P>

<P>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#AbstractSequence()">AbstractSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#AbstractSequence(org.biojava3.core.sequence.template.SequenceReader, org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;proxyLoader,
                 <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A ProxySequenceReader allows abstraction of both the storage of the sequence data and the location
 of the sequence data.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#AbstractSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</A></B>(java.lang.String&nbsp;seqString,
                 <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a Sequence from a simple string where the values should be found in compoundSet</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;feature)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add a feature to this sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(int, int, org.biojava3.core.sequence.features.FeatureInterface)">addFeature</A></B>(int&nbsp;bioStart,
           int&nbsp;bioEnd,
           <A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;feature)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Method to help set the proper details for a feature as it relates to a sequence
 where the feature needs to have a location on the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</A></B>(java.lang.String&nbsp;note)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add notes about this sequence that will get exported for GFF3</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>...&nbsp;compounds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the number of times we found a compound in the Sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the AccessionID this location is currently bound with</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the Sequence as a List of compounds</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Integer</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Integer</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</A></B>(int&nbsp;position)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the Compound at the given biological index</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Gets the compound set used to back this Sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</A></B>(int&nbsp;bioSequencePosition)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Return features at a sequence position</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String, int)">getFeatures</A></B>(java.lang.String&nbsp;featureType,
            int&nbsp;bioSequencePosition)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Return features at a sequence position by type</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</A></B>(java.lang.String&nbsp;type)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scans through the Sequence looking for the first occurrence of the given
 compound</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Does the <em>right thing</em> to get the inverse of the current
 Sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&nbsp;compound)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scans through the Sequence looking for the last occurrence of the given
 compound</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLength()">getLength</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the length of the Sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.ArrayList&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Default case is to assume strand is positive because only CDSSequence can be either positive or negative Strand.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer, java.lang.Integer, org.biojava3.core.sequence.Strand)">getSequenceAsString</A></B>(java.lang.Integer&nbsp;bioStart,
                    java.lang.Integer&nbsp;bioEnd,
                    <A HREF="../../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Provide place holder for a metric that indicate a score associated with the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSource()">getSource</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Added support for the source of this sequence for GFF3 export
 If a sub sequence doesn't have  source then check for parent source</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer, java.lang.Integer)">getSubSequence</A></B>(java.lang.Integer&nbsp;bioStart,
               java.lang.Integer&nbsp;bioEnd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a portion of the sequence from the different positions.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Collection&lt;java.lang.Object&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Provided for convience if the developer needs to associate data with a sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Iterator&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#iterator()">iterator</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;feature)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Remove a feature from the sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</A></B>(java.lang.String&nbsp;note)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A>&nbsp;annotationType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</A></B>(java.lang.Integer&nbsp;begin)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</A></B>(java.lang.Integer&nbsp;end)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</A>&nbsp;databaseReferences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</A></B>(java.lang.String&nbsp;description)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</A>&nbsp;featuresKeyWord)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</A></B>(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;notesList)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</A></B>(java.lang.String&nbsp;originalHeader)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;parentSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;proxyLoader)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Very important method that allows external mappings of sequence data and features.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</A></B>(java.lang.Double&nbsp;sequenceScore)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</A></B>(java.lang.String&nbsp;source)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Added support for the source of this sequence for GFF3 export</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</A></B>(<A HREF="../../../../../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</A>&nbsp;taxonomy)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</A></B>(java.util.Collection&lt;java.lang.Object&gt;&nbsp;userCollection)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>equals, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="AbstractSequence()"><!-- --></A><H3>
AbstractSequence</H3>
<PRE>
public <B>AbstractSequence</B>()</PRE>
<DL>
</DL>
<HR>

<A NAME="AbstractSequence(java.lang.String, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
AbstractSequence</H3>
<PRE>
public <B>AbstractSequence</B>(java.lang.String&nbsp;seqString,
                        <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</PRE>
<DL>
<DD>Create a Sequence from a simple string where the values should be found in compoundSet
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>seqString</CODE> - <DD><CODE>compoundSet</CODE> - </DL>
</DL>
<HR>

<A NAME="AbstractSequence(org.biojava3.core.sequence.template.SequenceReader, org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
AbstractSequence</H3>
<PRE>
public <B>AbstractSequence</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;proxyLoader,
                        <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</PRE>
<DL>
<DD>A ProxySequenceReader allows abstraction of both the storage of the sequence data and the location
 of the sequence data. A variety of use cases are possible. A ProxySequenceReader that knows the offset and of teh sequence in
 a large fasta file. A ProxySequenceReader that can pull Sequence data from Uniprot, NCBI or a custom database.
 If the ProxySequecneReader implements various interfaces then the sequence will set those interfaces so that calls to
 various methods will be valid.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>proxyLoader</CODE> - <DD><CODE>compoundSet</CODE> - </DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)"><!-- --></A><H3>
setProxySequenceReader</H3>
<PRE>
public void <B>setProxySequenceReader</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;proxyLoader)</PRE>
<DL>
<DD>Very important method that allows external mappings of sequence data and features. This method
 will gain additional interface inspection that allows external data sources with knowledge
 of features for a sequence to be supported.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>proxyLoader</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="getProxySequenceReader()"><!-- --></A><H3>
getProxySequenceReader</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt; <B>getProxySequenceReader</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getBioBegin()"><!-- --></A><H3>
getBioBegin</H3>
<PRE>
public java.lang.Integer <B>getBioBegin</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the bioBegin</DL>
</DD>
</DL>
<HR>

<A NAME="setBioBegin(java.lang.Integer)"><!-- --></A><H3>
setBioBegin</H3>
<PRE>
public void <B>setBioBegin</B>(java.lang.Integer&nbsp;begin)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>bioBegin</CODE> - the bioBegin to set</DL>
</DD>
</DL>
<HR>

<A NAME="getBioEnd()"><!-- --></A><H3>
getBioEnd</H3>
<PRE>
public java.lang.Integer <B>getBioEnd</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the bioEnd</DL>
</DD>
</DL>
<HR>

<A NAME="setBioEnd(java.lang.Integer)"><!-- --></A><H3>
setBioEnd</H3>
<PRE>
public void <B>setBioEnd</B>(java.lang.Integer&nbsp;end)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>bioEnd</CODE> - the bioEnd to set</DL>
</DD>
</DL>
<HR>

<A NAME="getUserCollection()"><!-- --></A><H3>
getUserCollection</H3>
<PRE>
public java.util.Collection&lt;java.lang.Object&gt; <B>getUserCollection</B>()</PRE>
<DL>
<DD>Provided for convience if the developer needs to associate data with a sequence
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="setUserCollection(java.util.Collection)"><!-- --></A><H3>
setUserCollection</H3>
<PRE>
public void <B>setUserCollection</B>(java.util.Collection&lt;java.lang.Object&gt;&nbsp;userCollection)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>userCollection</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="getAnnotationType()"><!-- --></A><H3>
getAnnotationType</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A> <B>getAnnotationType</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the annotation</DL>
</DD>
</DL>
<HR>

<A NAME="setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)"><!-- --></A><H3>
setAnnotationType</H3>
<PRE>
public void <B>setAnnotationType</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</A>&nbsp;annotationType)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>annotation</CODE> - the annotation to set</DL>
</DD>
</DL>
<HR>

<A NAME="getDescription()"><!-- --></A><H3>
getDescription</H3>
<PRE>
public java.lang.String <B>getDescription</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the description</DL>
</DD>
</DL>
<HR>

<A NAME="setDescription(java.lang.String)"><!-- --></A><H3>
setDescription</H3>
<PRE>
public void <B>setDescription</B>(java.lang.String&nbsp;description)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>description</CODE> - the description to set</DL>
</DD>
</DL>
<HR>

<A NAME="getOriginalHeader()"><!-- --></A><H3>
getOriginalHeader</H3>
<PRE>
public java.lang.String <B>getOriginalHeader</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the originalHeader</DL>
</DD>
</DL>
<HR>

<A NAME="setOriginalHeader(java.lang.String)"><!-- --></A><H3>
setOriginalHeader</H3>
<PRE>
public void <B>setOriginalHeader</B>(java.lang.String&nbsp;originalHeader)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>originalHeader</CODE> - the originalHeader to set</DL>
</DD>
</DL>
<HR>

<A NAME="getParentSequence()"><!-- --></A><H3>
getParentSequence</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt; <B>getParentSequence</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the parentSequence</DL>
</DD>
</DL>
<HR>

<A NAME="setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)"><!-- --></A><H3>
setParentSequence</H3>
<PRE>
public void <B>setParentSequence</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;parentSequence)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>parentSequence</CODE> - the parentSequence to set</DL>
</DD>
</DL>
<HR>

<A NAME="getSource()"><!-- --></A><H3>
getSource</H3>
<PRE>
public java.lang.String <B>getSource</B>()</PRE>
<DL>
<DD>Added support for the source of this sequence for GFF3 export
 If a sub sequence doesn't have  source then check for parent source
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the source</DL>
</DD>
</DL>
<HR>

<A NAME="setSource(java.lang.String)"><!-- --></A><H3>
setSource</H3>
<PRE>
public void <B>setSource</B>(java.lang.String&nbsp;source)</PRE>
<DL>
<DD>Added support for the source of this sequence for GFF3 export
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>source</CODE> - the source to set</DL>
</DD>
</DL>
<HR>

<A NAME="addNote(java.lang.String)"><!-- --></A><H3>
addNote</H3>
<PRE>
public void <B>addNote</B>(java.lang.String&nbsp;note)</PRE>
<DL>
<DD>Add notes about this sequence that will get exported for GFF3
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>note</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="removeNote(java.lang.String)"><!-- --></A><H3>
removeNote</H3>
<PRE>
public void <B>removeNote</B>(java.lang.String&nbsp;note)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getNotesList()"><!-- --></A><H3>
getNotesList</H3>
<PRE>
public java.util.ArrayList&lt;java.lang.String&gt; <B>getNotesList</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the notesList</DL>
</DD>
</DL>
<HR>

<A NAME="setNotesList(java.util.ArrayList)"><!-- --></A><H3>
setNotesList</H3>
<PRE>
public void <B>setNotesList</B>(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;notesList)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>notesList</CODE> - the notesList to set</DL>
</DD>
</DL>
<HR>

<A NAME="getSequenceScore()"><!-- --></A><H3>
getSequenceScore</H3>
<PRE>
public java.lang.Double <B>getSequenceScore</B>()</PRE>
<DL>
<DD>Provide place holder for a metric that indicate a score associated with the sequence
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the sequenceScore</DL>
</DD>
</DL>
<HR>

<A NAME="setSequenceScore(java.lang.Double)"><!-- --></A><H3>
setSequenceScore</H3>
<PRE>
public void <B>setSequenceScore</B>(java.lang.Double&nbsp;sequenceScore)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>sequenceScore</CODE> - the sequenceScore to set</DL>
</DD>
</DL>
<HR>

<A NAME="getFeatures(java.lang.String, int)"><!-- --></A><H3>
getFeatures</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt; <B>getFeatures</B>(java.lang.String&nbsp;featureType,
                                                                           int&nbsp;bioSequencePosition)</PRE>
<DL>
<DD>Return features at a sequence position by type
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>featureType</CODE> - <DD><CODE>bioSequencePosition</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getFeatures(int)"><!-- --></A><H3>
getFeatures</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt; <B>getFeatures</B>(int&nbsp;bioSequencePosition)</PRE>
<DL>
<DD>Return features at a sequence position
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>featureType</CODE> - <DD><CODE>bioSequencePosition</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getFeatures()"><!-- --></A><H3>
getFeatures</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt; <B>getFeatures</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="addFeature(int, int, org.biojava3.core.sequence.features.FeatureInterface)"><!-- --></A><H3>
addFeature</H3>
<PRE>
public void <B>addFeature</B>(int&nbsp;bioStart,
                       int&nbsp;bioEnd,
                       <A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;feature)</PRE>
<DL>
<DD>Method to help set the proper details for a feature as it relates to a sequence
 where the feature needs to have a location on the sequence
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>bioStart</CODE> - <DD><CODE>bioEnd</CODE> - <DD><CODE>feature</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="addFeature(org.biojava3.core.sequence.features.FeatureInterface)"><!-- --></A><H3>
addFeature</H3>
<PRE>
public void <B>addFeature</B>(<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;feature)</PRE>
<DL>
<DD>Add a feature to this sequence. The feature will be added to the collection where the order is start position and if more than
 one feature at the same start position then longest is added first. This helps on doing feature layout for displaying features
 in SequenceFeaturePanel
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>feature</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="removeFeature(org.biojava3.core.sequence.features.FeatureInterface)"><!-- --></A><H3>
removeFeature</H3>
<PRE>
public void <B>removeFeature</B>(<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;feature)</PRE>
<DL>
<DD>Remove a feature from the sequence
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>feature</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="getFeaturesByType(java.lang.String)"><!-- --></A><H3>
getFeaturesByType</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;,<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&gt; <B>getFeaturesByType</B>(java.lang.String&nbsp;type)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>type</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getFeaturesKeyWord()"><!-- --></A><H3>
getFeaturesKeyWord</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</A> <B>getFeaturesKeyWord</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the featuresKeyWord</DL>
</DD>
</DL>
<HR>

<A NAME="setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)"><!-- --></A><H3>
setFeaturesKeyWord</H3>
<PRE>
public void <B>setFeaturesKeyWord</B>(<A HREF="../../../../../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</A>&nbsp;featuresKeyWord)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>featuresKeyWord</CODE> - the featuresKeyWord to set</DL>
</DD>
</DL>
<HR>

<A NAME="getDatabaseReferences()"><!-- --></A><H3>
getDatabaseReferences</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</A> <B>getDatabaseReferences</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the databaseReferences</DL>
</DD>
</DL>
<HR>

<A NAME="setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)"><!-- --></A><H3>
setDatabaseReferences</H3>
<PRE>
public void <B>setDatabaseReferences</B>(<A HREF="../../../../../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</A>&nbsp;databaseReferences)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>databaseReferences</CODE> - the databaseReferences to set</DL>
</DD>
</DL>
<HR>

<A NAME="getAccession()"><!-- --></A><H3>
getAccession</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A> <B>getAccession</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Accessioned.html#getAccession()">Accessioned</A></CODE></B></DD>
<DD>Returns the AccessionID this location is currently bound with
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Accessioned.html#getAccession()">getAccession</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the accession</DL>
</DD>
</DL>
<HR>

<A NAME="setAccession(org.biojava3.core.sequence.AccessionID)"><!-- --></A><H3>
setAccession</H3>
<PRE>
public void <B>setAccession</B>(<A HREF="../../../../../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</A>&nbsp;accession)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>accession</CODE> - the accession to set</DL>
</DD>
</DL>
<HR>

<A NAME="getTaxonomy()"><!-- --></A><H3>
getTaxonomy</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</A> <B>getTaxonomy</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the species</DL>
</DD>
</DL>
<HR>

<A NAME="setTaxonomy(org.biojava3.core.sequence.TaxonomyID)"><!-- --></A><H3>
setTaxonomy</H3>
<PRE>
public void <B>setTaxonomy</B>(<A HREF="../../../../../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</A>&nbsp;taxonomy)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>species</CODE> - the species to set</DL>
</DD>
</DL>
<HR>

<A NAME="getCompoundSet()"><!-- --></A><H3>
getCompoundSet</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt; <B>getCompoundSet</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getCompoundSet()">Sequence</A></CODE></B></DD>
<DD>Gets the compound set used to back this Sequence
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getCompoundSet()">getCompoundSet</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)"><!-- --></A><H3>
setCompoundSet</H3>
<PRE>
public void <B>setCompoundSet</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt;&nbsp;compoundSet)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getSequenceAsString(java.lang.Integer, java.lang.Integer, org.biojava3.core.sequence.Strand)"><!-- --></A><H3>
getSequenceAsString</H3>
<PRE>
public java.lang.String <B>getSequenceAsString</B>(java.lang.Integer&nbsp;bioStart,
                                            java.lang.Integer&nbsp;bioEnd,
                                            <A HREF="../../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</A>&nbsp;strand)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>begin</CODE> - <DD><CODE>end</CODE> - <DD><CODE>strand</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getSequenceAsString()"><!-- --></A><H3>
getSequenceAsString</H3>
<PRE>
public java.lang.String <B>getSequenceAsString</B>()</PRE>
<DL>
<DD>Default case is to assume strand is positive because only CDSSequence can be either positive or negative Strand.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getSequenceAsString()">getSequenceAsString</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getAsList()"><!-- --></A><H3>
getAsList</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt; <B>getAsList</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getAsList()">Sequence</A></CODE></B></DD>
<DD>Returns the Sequence as a List of compounds
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getAsList()">getAsList</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getCompoundAt(int)"><!-- --></A><H3>
getCompoundAt</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> <B>getCompoundAt</B>(int&nbsp;position)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getCompoundAt(int)">Sequence</A></CODE></B></DD>
<DD>Returns the Compound at the given biological index
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getCompoundAt(int)">getCompoundAt</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>position</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getIndexOf(org.biojava3.core.sequence.template.Compound)"><!-- --></A><A NAME="getIndexOf(C)"><!-- --></A><H3>
getIndexOf</H3>
<PRE>
public int <B>getIndexOf</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&nbsp;compound)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getIndexOf(C)">Sequence</A></CODE></B></DD>
<DD>Scans through the Sequence looking for the first occurrence of the given
 compound
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getIndexOf(C)">getIndexOf</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>compound</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getLastIndexOf(org.biojava3.core.sequence.template.Compound)"><!-- --></A><A NAME="getLastIndexOf(C)"><!-- --></A><H3>
getLastIndexOf</H3>
<PRE>
public int <B>getLastIndexOf</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&nbsp;compound)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getLastIndexOf(C)">Sequence</A></CODE></B></DD>
<DD>Scans through the Sequence looking for the last occurrence of the given
 compound
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getLastIndexOf(C)">getLastIndexOf</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>compound</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getLength()"><!-- --></A><H3>
getLength</H3>
<PRE>
public int <B>getLength</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getLength()">Sequence</A></CODE></B></DD>
<DD>Returns the length of the Sequence
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getLength()">getLength</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getSubSequence(java.lang.Integer, java.lang.Integer)"><!-- --></A><H3>
getSubSequence</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt; <B>getSubSequence</B>(java.lang.Integer&nbsp;bioStart,
                                      java.lang.Integer&nbsp;bioEnd)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getSubSequence(java.lang.Integer, java.lang.Integer)">Sequence</A></CODE></B></DD>
<DD>Returns a portion of the sequence from the different positions. This is
 indexed from 1
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getSubSequence(java.lang.Integer, java.lang.Integer)">getSubSequence</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>bioStart</CODE> - <DD><CODE>bioEnd</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="iterator()"><!-- --></A><H3>
iterator</H3>
<PRE>
public java.util.Iterator&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt; <B>iterator</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>iterator</CODE> in interface <CODE>java.lang.Iterable&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="countCompounds(org.biojava3.core.sequence.template.Compound[])"><!-- --></A><A NAME="countCompounds(C...)"><!-- --></A><H3>
countCompounds</H3>
<PRE>
public int <B>countCompounds</B>(<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>...&nbsp;compounds)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#countCompounds(C...)">Sequence</A></CODE></B></DD>
<DD>Returns the number of times we found a compound in the Sequence
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#countCompounds(C...)">countCompounds</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>compounds</CODE> - 
<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getInverse()"><!-- --></A><H3>
getInverse</H3>
<PRE>
public <A HREF="../../../../../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A>&gt; <B>getInverse</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getInverse()">Sequence</A></CODE></B></DD>
<DD>Does the <em>right thing</em> to get the inverse of the current
 Sequence. This means either reversing the Sequence and optionally
 complementing the Sequence.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html#getInverse()">getInverse</A></CODE> in interface <CODE><A HREF="../../../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</A>&lt;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</A> extends <A HREF="../../../../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</A>&gt;</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/AbstractSequence.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/biojava3/core/sequence/template/AbstractSequence.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="AbstractSequence.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
